*** ../generated-files/_remove_nulls_in_array_klasse.sql	2015-06-12 11:02:39.500538510 +0200
--- ../patches/_remove_nulls_in_array_klasse.org.sql	2015-06-12 11:06:28.520549916 +0200
***************
*** 49,59 ****
    BEGIN
  
    IF inputArr IS NOT NULL THEN
      FOREACH element IN ARRAY  inputArr
      LOOP
!       IF element IS NULL OR (( element.brugervendtnoegle IS NULL AND element.beskrivelse IS NULL AND element.eksempel IS NULL AND element.omfang IS NULL AND element.titel IS NULL AND element.retskilde IS NULL AND element.aendringsnotat IS NULL ) AND element.virkning IS NULL) THEN --CAUTION: foreach on {null} will result in element gets initiated with ROW(null,null....) 
      --  RAISE DEBUG 'Skipping element';
        ELSE
        result:=array_append(result,element);
        END IF;
      END LOOP;
--- 49,59 ----
    BEGIN
  
    IF inputArr IS NOT NULL THEN
      FOREACH element IN ARRAY  inputArr
      LOOP
!       IF element IS NULL OR (( element.brugervendtnoegle IS NULL AND element.beskrivelse IS NULL AND element.eksempel IS NULL AND element.omfang IS NULL AND element.titel IS NULL AND element.retskilde IS NULL AND element.aendringsnotat IS NULL ) AND element.virkning IS NULL AND (element.soegeord IS NULL OR coalesce(array_length(element.soegeord,1),0)=0 )) THEN --CAUTION: foreach on {null} will result in element gets initiated with ROW(null,null....) 
      --  RAISE DEBUG 'Skipping element';
        ELSE
        result:=array_append(result,element);
        END IF;
      END LOOP;
***************
*** 97,102 ****
--- 97,132 ----
   
   $$ LANGUAGE plpgsql IMMUTABLE
  ;
  
  
+ CREATE OR REPLACE FUNCTION _remove_nulls_in_array(inputArr KlasseSoegeordType[])
+   RETURNS KlasseSoegeordType[] AS
+   $$
+   DECLARE result KlasseSoegeordType[];
+   DECLARE element KlasseSoegeordType;
+   BEGIN
+ 
+  IF inputArr IS NOT NULL THEN
+     FOREACH element IN ARRAY  inputArr
+     LOOP
+       IF element IS NULL OR (element.soegeordidentifikator IS NULL AND element.beskrivelse IS NULL AND element.soegeordskategori IS NULL ) THEN
+      -- RAISE DEBUG 'Skipping element';
+       ELSE
+       result:=array_append(result,element);
+       END IF;
+     END LOOP;
+   ELSE
+     return null;  
+   END IF;
+ 
+   IF array_length(result,1)=0 THEN
+     RETURN NULL;
+   ELSE
+     RETURN result;
+   END IF;
+ 
+   END;
+  
+  $$ LANGUAGE plpgsql IMMUTABLE
+ ;
  
