*** ../generated-files/as_create_or_import_klasse.sql	2015-05-26 14:43:39.547814798 +0200
--- ../patches/as_create_or_import_klasse_basis.sql	2015-05-26 15:24:54.015938034 +0200
***************
*** 6,12 ****
  -- file, You can obtain one at http://mozilla.org/MPL/2.0/.
  
  /*
! NOTICE: This file is auto-generated using the script: apply-template.py klasse as_create_or_import.jinja.sql
  */
  
  CREATE OR REPLACE FUNCTION as_create_or_import_klasse(
--- 6,12 ----
  -- file, You can obtain one at http://mozilla.org/MPL/2.0/.
  
  /*
! NOTICE: This file is auto-generated using the script: apply-template.py klasse as_create_or_import.jinja.sql AND applying a patch (as_create_or_import_klasse.sql.diff)
  */
  
  CREATE OR REPLACE FUNCTION as_create_or_import_klasse(
***************
*** 24,29 ****
--- 24,30 ----
    klasse_relationer KlasseRelationType;
  
    klasse_attr_egenskaber_id bigint;
+   klasse_attr_egenskaber_soegeord_obj KlasseSoegeordType;
  BEGIN
  
  IF EXISTS (SELECT id from klasse WHERE id=klasse_uuid) THEN
***************
*** 83,91 ****
    FOREACH klasse_attr_egenskaber_obj IN ARRAY klasse_registrering.attrEgenskaber
    LOOP
  
! 
  
    INSERT INTO klasse_attr_egenskaber (
      brugervendtnoegle,
      beskrivelse,
      eksempel,
--- 84,93 ----
    FOREACH klasse_attr_egenskaber_obj IN ARRAY klasse_registrering.attrEgenskaber
    LOOP
  
!   klasse_attr_egenskaber_id=nextval('klasse_attr_egenskaber_id_seq');
  
    INSERT INTO klasse_attr_egenskaber (
+     id,
      brugervendtnoegle,
      beskrivelse,
      eksempel,
***************
*** 97,102 ****
--- 99,105 ----
      klasse_registrering_id
    )
    SELECT
+     klasse_attr_egenskaber_id,
     klasse_attr_egenskaber_obj.brugervendtnoegle,
      klasse_attr_egenskaber_obj.beskrivelse,
      klasse_attr_egenskaber_obj.eksempel,
***************
*** 108,113 ****
--- 111,137 ----
      klasse_registrering_id
    ;
  
+ /************/
+ --Insert Soegeord
+   IF (klasse_registrering.attrEgenskaber).soegeord IS NOT NULL THEN
+     FOREACH klasse_attr_egenskaber_soegeord_obj IN ARRAY (klasse_registrering.attrEgenskaber).soegeord 
+       LOOP
+ 
+       INSERT INTO klasse_attr_egenskaber_soegeord (
+         soegeordidentifikator,
+         beskrivelse,
+         soegeordskategori,
+         klasse_attr_egenskaber_id,
+       )
+       SELECT
+         klasse_attr_egenskaber_soegeord_obj.soegeordidentifikator,
+         klasse_attr_egenskaber_soegeord_obj.beskrivelse,
+         klasse_attr_egenskaber_soegeord_obj.soegeordskategori,
+         klasse_attr_egenskaber_id
+       ;
+ 
+      END LOOP;
+     END IF
    END LOOP;
  END IF;
  
