*** ../generated-files/as_list_klasse.sql	2015-05-27 10:02:04.186215827 +0200
--- ../patches/as_list_klasse.sql.org	2015-05-27 10:28:40.622295334 +0200
***************
*** 59,100 ****
  						) ::KlassePubliceretTilsType
  					order by b.id
  				) KlasseTilsPubliceretArr		
  			FROM
  			(
! 					SELECT
! 					a.klasse_id,
! 					a.klasse_registrering_id,
! 					a.registrering,
! 					array_agg(
! 						ROW(
! 					 		b.brugervendtnoegle,
! 					 		b.beskrivelse,
! 					 		b.eksempel,
! 					 		b.omfang,
! 					 		b.titel,
! 					 		b.retskilde,
! 					 		b.aendringsnotat,
! 					   		b.virkning 
! 							)::KlasseEgenskaberAttrType
! 						order by b.id
! 					) KlasseAttrEgenskaberArr 
! 					FROM
! 					(
! 					SELECT
! 					a.id klasse_id,
! 					b.id klasse_registrering_id,
! 					b.registrering			
! 					FROM		klasse a
! 					JOIN 		klasse_registrering b 	ON b.klasse_id=a.id
! 					WHERE a.id = ANY (klasse_uuids) AND (((registrering_tstzrange is null OR isempty(registrering_tstzrange)) AND upper((b.registrering).timeperiod)='infinity'::TIMESTAMPTZ) OR registrering_tstzrange && (b.registrering).timeperiod)--filter ON registrering_tstzrange
! 					) as a
! 					LEFT JOIN klasse_attr_egenskaber as b ON b.klasse_registrering_id=a.klasse_registrering_id AND ((virkning_tstzrange is null OR isempty(virkning_tstzrange)) OR (b.virkning).TimePeriod && virkning_tstzrange) --filter ON virkning_tstzrange if given			
! 					GROUP BY 
! 					a.klasse_id,
! 					a.klasse_registrering_id,
! 					a.registrering	
  			) as a
  			LEFT JOIN klasse_tils_publiceret as b ON b.klasse_registrering_id=a.klasse_registrering_id AND ((virkning_tstzrange is null OR isempty(virkning_tstzrange)) OR (b.virkning).TimePeriod && virkning_tstzrange) --filter ON virkning_tstzrange if given			
  			GROUP BY 
  			a.klasse_id,
  			a.klasse_registrering_id,
--- 59,139 ----
  						) ::KlassePubliceretTilsType
  					order by b.id
  				) KlasseTilsPubliceretArr		
  			FROM
  			(
! 				SELECT
! 				a.klasse_id,
! 				a.klasse_registrering_id,
! 				a.registrering,
! 				array_agg(
! 					ROW(
! 				 		a.brugervendtnoegle,
! 				 		a.beskrivelse,
! 				 		a.eksempel,
! 				 		a.omfang,
! 				 		a.titel,
! 				 		a.retskilde,
! 				 		a.aendringsnotat,
! 				 		a.KlasseAttrEgenskaberSoegeordTypeArr,
! 				   		a.virkning 
! 						)::KlasseEgenskaberAttrType
! 					order by a.attr_id
! 				) KlasseAttrEgenskaberArr 
! 				FROM		
! 				(
! 						SELECT
! 						a.klasse_id,
! 						a.klasse_registrering_id,
! 						a.registrering,
! 						b.id attr_id,
! 						b.brugervendtnoegle,
! 						b.beskrivelse,
! 						b.eksempel,
! 						b.omfang,
! 						b.titel,
! 						b.retskilde,
! 						b.aendringsnotat,
! 						b.virkning,	
! 						array_agg(
! 							ROW(
! 						 		c.soegeordidentifikator,
! 						 		c.beskrivelse,
! 						 		c.soegeordskategori 
! 								)::KlasseSoegeordType
! 							order by c.id
! 						) KlasseAttrEgenskaberSoegeordTypeArr
! 						FROM
! 						(
! 								SELECT
! 								a.id klasse_id,
! 								b.id klasse_registrering_id,
! 								b.registrering			
! 								FROM		klasse a
! 								JOIN 		klasse_registrering b 	ON b.klasse_id=a.id
! 								WHERE a.id = ANY (klasse_uuids) AND (((registrering_tstzrange is null OR isempty(registrering_tstzrange)) AND upper((b.registrering).timeperiod)='infinity'::TIMESTAMPTZ) OR registrering_tstzrange && (b.registrering).timeperiod)--filter ON registrering_tstzrange
! 							) as a
! 						LEFT JOIN klasse_attr_egenskaber as b ON b.klasse_registrering_id=a.klasse_registrering_id AND ((virkning_tstzrange is null OR isempty(virkning_tstzrange)) OR (b.virkning).TimePeriod && virkning_tstzrange) --filter ON virkning_tstzrange if given			
! 						LEFT JOIN klasse_attr_egenskaber_soegeord as c ON c.klasse_attr_egenskaber_id=b.id
! 						GROUP BY 
! 						a.klasse_id,
! 						a.klasse_registrering_id,
! 						a.registrering,
! 						b.id,
! 						b.brugervendtnoegle,
! 						b.beskrivelse,
! 						b.eksempel,
! 						b.omfang,
! 						b.titel,
! 						b.retskilde,
! 						b.aendringsnotat,
! 						b.virkning
! 				) as a
! 			GROUP BY 
! 			a.klasse_id,
! 			a.klasse_registrering_id,
! 			a.registrering
  			) as a
  			LEFT JOIN klasse_tils_publiceret as b ON b.klasse_registrering_id=a.klasse_registrering_id AND ((virkning_tstzrange is null OR isempty(virkning_tstzrange)) OR (b.virkning).TimePeriod && virkning_tstzrange) --filter ON virkning_tstzrange if given			
  			GROUP BY 
  			a.klasse_id,
  			a.klasse_registrering_id,
