*** ../generated-files/as_update_klasse.sql	2015-05-27 12:01:12.298571824 +0200
--- ../patches/as_update_klasse.org.sql	2015-05-27 15:11:57.735141841 +0200
***************
*** 30,39 ****
--- 30,41 ----
  DECLARE
    new_klasse_registrering klasse_registrering;
    prev_klasse_registrering klasse_registrering;
    klasse_relation_navn KlasseRelationKode;
    attrEgenskaberObj KlasseEgenskaberAttrType;
+   new_id_klasse_attr_egenskaber bigint;
+   klasseSoegeordObj KlasseSoegeordType;
  BEGIN
  
  --create a new registrering
  
  new_klasse_registrering := _as_create_klasse_registrering(klasse_uuid,livscykluskode, brugerref, note);
***************
*** 216,233 ****
     (attrEgenskaberObj).titel is null OR 
     (attrEgenskaberObj).retskilde is null OR 
     (attrEgenskaberObj).aendringsnotat is null 
    THEN
  
    INSERT INTO
    klasse_attr_egenskaber
    (
!     brugervendtnoegle,beskrivelse,eksempel,omfang,titel,retskilde,aendringsnotat
      ,virkning
      ,klasse_registrering_id
    )
    SELECT 
      coalesce(attrEgenskaberObj.brugervendtnoegle,a.brugervendtnoegle), 
      coalesce(attrEgenskaberObj.beskrivelse,a.beskrivelse), 
      coalesce(attrEgenskaberObj.eksempel,a.eksempel), 
      coalesce(attrEgenskaberObj.omfang,a.omfang), 
      coalesce(attrEgenskaberObj.titel,a.titel), 
--- 218,237 ----
     (attrEgenskaberObj).titel is null OR 
     (attrEgenskaberObj).retskilde is null OR 
     (attrEgenskaberObj).aendringsnotat is null 
    THEN
  
+ WITH inserted_merged_attr_egenskaber AS (
    INSERT INTO
    klasse_attr_egenskaber
    (
!     id,brugervendtnoegle,beskrivelse,eksempel,omfang,titel,retskilde,aendringsnotat
      ,virkning
      ,klasse_registrering_id
    )
    SELECT 
+     nextval('klasse_attr_egenskaber_id_seq'),
      coalesce(attrEgenskaberObj.brugervendtnoegle,a.brugervendtnoegle), 
      coalesce(attrEgenskaberObj.beskrivelse,a.beskrivelse), 
      coalesce(attrEgenskaberObj.eksempel,a.eksempel), 
      coalesce(attrEgenskaberObj.omfang,a.omfang), 
      coalesce(attrEgenskaberObj.titel,a.titel), 
***************
*** 242,263 ****
      new_klasse_registrering.id
    FROM klasse_attr_egenskaber a
    WHERE
      a.klasse_registrering_id=prev_klasse_registrering.id 
      and (a.virkning).TimePeriod && (attrEgenskaberObj.virkning).TimePeriod
!   ;
  
    --For any periods within the virkning of the attrEgenskaberObj, that is NOT covered by any "merged" rows inserted above, generate and insert rows
  
    INSERT INTO
    klasse_attr_egenskaber
    (
!     brugervendtnoegle,beskrivelse,eksempel,omfang,titel,retskilde,aendringsnotat
      ,virkning
      ,klasse_registrering_id
    )
    SELECT 
      attrEgenskaberObj.brugervendtnoegle, 
      attrEgenskaberObj.beskrivelse, 
      attrEgenskaberObj.eksempel, 
      attrEgenskaberObj.omfang, 
      attrEgenskaberObj.titel, 
--- 246,282 ----
      new_klasse_registrering.id
    FROM klasse_attr_egenskaber a
    WHERE
      a.klasse_registrering_id=prev_klasse_registrering.id 
      and (a.virkning).TimePeriod && (attrEgenskaberObj.virkning).TimePeriod
!     RETURNING id new_id,a.id merged_with_id 
! )
! INSERT INTO 
! klasse_attr_egenskaber_soegeord 
! (soegeordidentifikator,beskrivelse,soegeordskategori,klasse_attr_egenskaber_id)
! SELECT
!   coalesce(b.soegeordidentifikator,c.soegeordidentifikator), --please notice that this is not a merge - one of the joins on b or c will fail.
!   coalesce(b.beskrivelse,c.beskrivelse),--please notice that this is not a merge - one of the joins on b or c will fail.
!   coalesce(b.soegeordskategori,c.soegeordskategori),--please notice that this is not a merge - one of the joins on b or c will fail.
!   a.new_id
! FROM inserted_merged_attr_egenskaber a
! LEFT JOIN unnest(attrEgenskaberObj.soegeord) as b(soegeordidentifikator,beskrivelse,soegeordskategori) on attrEgenskaberObj.soegeord IS NOT NULL
! LEFT JOIN klasse_attr_egenskaber_soegeord as c on attrEgenskaberObj.soegeord IS NULL AND c.klasse_attr_egenskaber_id = a.merged_with_id
! ;
  
    --For any periods within the virkning of the attrEgenskaberObj, that is NOT covered by any "merged" rows inserted above, generate and insert rows
  
+ WITH inserted_attr_egenskaber AS (
    INSERT INTO
    klasse_attr_egenskaber
    (
!     id,brugervendtnoegle,beskrivelse,eksempel,omfang,titel,retskilde,aendringsnotat
      ,virkning
      ,klasse_registrering_id
    )
    SELECT 
+     nextval('klasse_attr_egenskaber_id_seq'),
      attrEgenskaberObj.brugervendtnoegle, 
      attrEgenskaberObj.beskrivelse, 
      attrEgenskaberObj.eksempel, 
      attrEgenskaberObj.omfang, 
      attrEgenskaberObj.titel, 
***************
*** 277,323 ****
        FROM klasse_attr_egenskaber b
        WHERE 
         b.klasse_registrering_id=new_klasse_registrering.id
    ) as a
    JOIN unnest(_subtract_tstzrange_arr((attrEgenskaberObj.virkning).TimePeriod,a.tzranges_of_new_reg)) as b(tz_range_leftover) on true
!   ;
  
    ELSE
      --insert attrEgenskaberObj raw (if there were no null-valued fields) 
  
      INSERT INTO
      klasse_attr_egenskaber
      (
!     brugervendtnoegle,beskrivelse,eksempel,omfang,titel,retskilde,aendringsnotat
      ,virkning
      ,klasse_registrering_id
      )
      VALUES ( 
      attrEgenskaberObj.brugervendtnoegle, 
      attrEgenskaberObj.beskrivelse, 
      attrEgenskaberObj.eksempel, 
      attrEgenskaberObj.omfang, 
      attrEgenskaberObj.titel, 
      attrEgenskaberObj.retskilde, 
      attrEgenskaberObj.aendringsnotat,
      attrEgenskaberObj.virkning,
      new_klasse_registrering.id
!     );
  
    END IF;
  
    END LOOP;
  END IF;
  
  --Handle egenskaber of previous registration, taking overlapping virknings into consideration (using function subtract_tstzrange)
  
  INSERT INTO klasse_attr_egenskaber (
!     brugervendtnoegle,beskrivelse,eksempel,omfang,titel,retskilde,aendringsnotat
      ,virkning
      ,klasse_registrering_id
  )
  SELECT
        a.brugervendtnoegle,
        a.beskrivelse,
        a.eksempel,
        a.omfang,
        a.titel,
--- 296,380 ----
        FROM klasse_attr_egenskaber b
        WHERE 
         b.klasse_registrering_id=new_klasse_registrering.id
    ) as a
    JOIN unnest(_subtract_tstzrange_arr((attrEgenskaberObj.virkning).TimePeriod,a.tzranges_of_new_reg)) as b(tz_range_leftover) on true
!   RETURNING id
!   )
! INSERT INTO 
! klasse_attr_egenskaber_soegeord 
! (soegeordidentifikator,beskrivelse,soegeordskategori,klasse_attr_egenskaber_id)
! SELECT
! a.soegeordidentifikator,a.beskrivelse,a.soegeordskategori,b.id
! FROM
! unnest(attrEgenskaberObj.soegeord) as a(soegeordidentifikator,beskrivelse,soegeordskategori)
! JOIN inserted_attr_egenskaber b on true
! ;
! 
! 
  
    ELSE
      --insert attrEgenskaberObj raw (if there were no null-valued fields) 
  
+     new_id_klasse_attr_egenskaber:=nextval('klasse_attr_egenskaber_id_seq');
+ 
      INSERT INTO
      klasse_attr_egenskaber
      (
!     id,brugervendtnoegle,beskrivelse,eksempel,omfang,titel,retskilde,aendringsnotat
      ,virkning
      ,klasse_registrering_id
      )
      VALUES ( 
+     new_id_klasse_attr_egenskaber,
      attrEgenskaberObj.brugervendtnoegle, 
      attrEgenskaberObj.beskrivelse, 
      attrEgenskaberObj.eksempel, 
      attrEgenskaberObj.omfang, 
      attrEgenskaberObj.titel, 
      attrEgenskaberObj.retskilde, 
      attrEgenskaberObj.aendringsnotat,
      attrEgenskaberObj.virkning,
      new_klasse_registrering.id
!     )
!     ;
!    
!     IF attrEgenskaberObj.soegeord IS NOT NULL THEN
!     INSERT INTO klasse_attr_egenskaber_soegeord( 
!           soegeordidentifikator,
!           beskrivelse,
!           soegeordskategori,
!           klasse_attr_egenskaber_id
!           )
!     SELECT
!     a.soegeordidentifikator,
!     a.beskrivelse,
!     a.soegeordskategori,
!     new_id_klasse_attr_egenskaber
!     FROM
!     unnest(attrEgenskaberObj.soegeord) as a(soegeordidentifikator,beskrivelse,soegeordskategori)
!     ;
!     END IF;
! 
  
    END IF;
  
    END LOOP;
  END IF;
  
  --Handle egenskaber of previous registration, taking overlapping virknings into consideration (using function subtract_tstzrange)
  
+ 
+ 
+ WITH copied_attr_egenskaber AS (
  INSERT INTO klasse_attr_egenskaber (
!     id,brugervendtnoegle,beskrivelse,eksempel,omfang,titel,retskilde,aendringsnotat
      ,virkning
      ,klasse_registrering_id
  )
  SELECT
+       nextval('klasse_attr_egenskaber_id_seq'),
        a.brugervendtnoegle,
        a.beskrivelse,
        a.eksempel,
        a.omfang,
        a.titel,
***************
*** 338,348 ****
      WHERE 
            b.klasse_registrering_id=new_klasse_registrering.id
  ) d
    JOIN klasse_attr_egenskaber a ON true  
    JOIN unnest(_subtract_tstzrange_arr((a.virkning).TimePeriod,tzranges_of_new_reg)) as c(tz_range_leftover) on true
!   WHERE a.klasse_registrering_id=prev_klasse_registrering.id     
  ;
  
  
  return new_klasse_registrering.id;
  
--- 395,414 ----
      WHERE 
            b.klasse_registrering_id=new_klasse_registrering.id
  ) d
    JOIN klasse_attr_egenskaber a ON true  
    JOIN unnest(_subtract_tstzrange_arr((a.virkning).TimePeriod,tzranges_of_new_reg)) as c(tz_range_leftover) on true
!   WHERE a.klasse_registrering_id=prev_klasse_registrering.id 
!   RETURNING id new_id,a.id based_on_id
! )
! INSERT INTO 
! klasse_attr_egenskaber_soegeord 
! (soegeordidentifikator,beskrivelse,soegeordskategori,klasse_attr_egenskaber_id)
! SELECT
! b.soegeordidentifikator,b.beskrivelse,b.soegeordskategori,a.new_id
! FROM copied_attr_egenskaber a
! JOIN klasse_attr_egenskaber_soegeord b on a.based_on_id=b.klasse_attr_egenskaber_id   
  ;
  
  
  return new_klasse_registrering.id;
  
***************
*** 350,357 ****
  
  END;
  $$ LANGUAGE plpgsql VOLATILE;
  
  
- 
- 
- 
--- 416,420 ----
