*** ../generated-files/dbtyper-specific_klasse.sql	2015-05-26 15:33:17.675963118 +0200
--- ../patches/dbtyper-specific_klasse_basis.sql	2015-05-26 15:04:30.419877095 +0200
***************
*** 6,12 ****
  -- file, You can obtain one at http://mozilla.org/MPL/2.0/.
  
  /*
! NOTICE: This file is auto-generated using the script: apply-template.py klasse dbtyper-specific.jinja.sql
  */
  
  --create custom type sans db-ids to be able to do "clean" function signatures "for the outside world".
--- 6,12 ----
  -- file, You can obtain one at http://mozilla.org/MPL/2.0/.
  
  /*
! NOTICE: This file is auto-generated using the script: apply-template.py klasse dbtyper-specific.jinja.sql AND applying a patch (dbtyper-specific_klasse.sql.diff)
  */
  
  --create custom type sans db-ids to be able to do "clean" function signatures "for the outside world".
***************
*** 19,24 ****
--- 19,31 ----
  )
  ;
  
+ CREATE TYPE KlasseSoegeordType AS (
+ soegeordidentifikator text,
+ beskrivelse text,
+ soegeordskategori text
+ )
+ ;
+ 
  CREATE TYPE KlasseEgenskaberAttrType AS (
  brugervendtnoegle text,
  beskrivelse text,
***************
*** 27,32 ****
--- 34,40 ----
  titel text,
  retskilde text,
  aendringsnotat text,
+ soegeord KlasseSoegeordType[],
   virkning Virkning
  );
  
