*** ./generated-files/_remove_nulls_in_array_tilstand.sql	2016-10-22 10:03:00.000000000 +0200
--- patches/_remove_nulls_in_array_tilstand.org.sql	2016-10-22 09:44:34.000000000 +0200
***************
*** 107,117 ****
    BEGIN
  
     IF inputArr IS NOT NULL THEN
      FOREACH element IN ARRAY  inputArr
      LOOP
!       IF element IS NULL OR ( element.relType IS NULL AND element.uuid IS NULL AND element.urn IS NULL AND element.objektType IS NULL AND element.virkning IS NULL  ) THEN --CAUTION: foreach on {null} will result in element gets initiated with ROW(null,null....) 
        --RAISE DEBUG 'Skipping element';
        ELSE
        result:=array_append(result,element);
        END IF;
      END LOOP;
--- 107,117 ----
    BEGIN
  
     IF inputArr IS NOT NULL THEN
      FOREACH element IN ARRAY  inputArr
      LOOP
!       IF element IS NULL OR ( element.relType IS NULL AND element.uuid IS NULL AND element.urn IS NULL AND element.objektType IS NULL AND element.indeks IS NULL AND (element.tilstandsVaerdiAttr IS NULL OR ((element.tilstandsVaerdiAttr).nominelVaerdi IS NULL AND (element.tilstandsVaerdiAttr).forventet IS NULL )) AND element.virkning IS NULL  ) THEN --CAUTION: foreach on {null} will result in element gets initiated with ROW(null,null....) 
        --RAISE DEBUG 'Skipping element';
        ELSE
        result:=array_append(result,element);
        END IF;
      END LOOP;
