*** ../generated-files/_remove_nulls_in_array_sag.sql	2015-06-24 11:56:17.333302348 +0200
--- ../patches/_remove_nulls_in_array_sag.org.sql	2015-06-24 12:59:35.537234330 +0200
***************
*** 79,89 ****
    BEGIN
  
     IF inputArr IS NOT NULL THEN
      FOREACH element IN ARRAY  inputArr
      LOOP
!       IF element IS NULL OR ( element.relType IS NULL AND element.relMaalUuid IS NULL AND element.relMaalUrn IS NULL AND element.objektType IS NULL AND element.virkning IS NULL  ) THEN --CAUTION: foreach on {null} will result in element gets initiated with ROW(null,null....) 
        --RAISE DEBUG 'Skipping element';
        ELSE
        result:=array_append(result,element);
        END IF;
      END LOOP;
--- 79,89 ----
    BEGIN
  
     IF inputArr IS NOT NULL THEN
      FOREACH element IN ARRAY  inputArr
      LOOP
!       IF element IS NULL OR ( element.relType IS NULL AND element.relMaalUuid IS NULL AND element.relMaalUrn IS NULL AND element.objektType IS NULL AND element.relIndex IS NULL AND element.relTypeSpec IS NULL AND (element.journalNotat IS NULL OR ( (element.journalNotat).titel IS NULL AND (element.journalNotat).notat IS NULL AND (element.journalNotat).format IS NULL )) AND (element.journalDokumentAttr IS NULL OR ((element.journalDokumentAttr).dokumenttitel IS NULL AND (element.journalDokumentAttr).offentlighedUndtaget IS NULL )) AND element.virkning IS NULL  ) THEN --CAUTION: foreach on {null} will result in element gets initiated with ROW(null,null....) 
        --RAISE DEBUG 'Skipping element';
        ELSE
        result:=array_append(result,element);
        END IF;
      END LOOP;
