*** ../generated-files/as_create_or_import_sag.sql	2015-08-06 09:54:58.776725516 +0200
--- ../patches/as_create_or_import_sag.org.sql	2015-08-06 09:57:53.732723550 +0200
***************
*** 20,30 ****
    sag_attr_egenskaber_obj sagEgenskaberAttrType;
    
    sag_tils_fremdrift_obj sagFremdriftTilsType;
    
    sag_relationer SagRelationType;
! 
  BEGIN
  
  IF sag_uuid IS NULL THEN
      LOOP
      sag_uuid:=uuid_generate_v4();
--- 20,33 ----
    sag_attr_egenskaber_obj sagEgenskaberAttrType;
    
    sag_tils_fremdrift_obj sagFremdriftTilsType;
    
    sag_relationer SagRelationType;
!   sag_relation_kode SagRelationKode;
!   sag_uuid_underscores text;
!   sag_rel_seq_name text;
!   sag_rel_type_cardinality_unlimited SagRelationKode[]:=ARRAY['andetarkiv'::SagRelationKode,'andrebehandlere'::SagRelationKode,'sekundaerpart'::SagRelationKode,'andresager'::SagRelationKode,'byggeri'::SagRelationKode,'fredning'::SagRelationKode,'journalpost'::SagRelationKode]::SagRelationKode[];
  BEGIN
  
  IF sag_uuid IS NULL THEN
      LOOP
      sag_uuid:=uuid_generate_v4();
***************
*** 172,200 ****
  END IF;
  
  /*********************************/
  --Insert relations
  
      INSERT INTO sag_relation (
        sag_registrering_id,
        virkning,
        rel_maal_uuid,
        rel_maal_urn,
        rel_type,
!       objekt_type
      )
      SELECT
        sag_registrering_id,
        a.virkning,
        a.relMaalUuid,
        a.relMaalUrn,
        a.relType,
!       a.objektType
      FROM unnest(sag_registrering.relationer) a
!     WHERE (a.relMaalUuid IS NOT NULL OR (a.relMaalUrn IS NOT NULL AND a.relMaalUrn<>'') )
    ;
  
    PERFORM amqp.publish(1, 'mox.notifications', '', format('create %s', sag_uuid));
  
  RETURN sag_uuid;
  
  END;
--- 175,254 ----
  END IF;
  
  /*********************************/
  --Insert relations
  
+ IF coalesce(array_length(sag_registrering.relationer,1),0)>0 THEN
+ 
+ --Create temporary sequences
+ sag_uuid_underscores:=replace(sag_uuid::text, '-', '_');
+ 
+ FOREACH sag_relation_kode IN ARRAY (SELECT array_agg( DISTINCT a.RelType) FROM  unnest(sag_registrering.relationer) a WHERE a.RelType = any (sag_rel_type_cardinality_unlimited))
+   LOOP
+   sag_rel_seq_name := 'sag_rel_' || sag_relation_kode::text || sag_uuid_underscores;
+ 
+   EXECUTE 'CREATE TEMPORARY SEQUENCE ' || sag_rel_seq_name || '
+   INCREMENT 1
+   MINVALUE 1
+   MAXVALUE 9223372036854775807
+   START 1
+   CACHE 1;';
+ 
+ END LOOP;
+ 
      INSERT INTO sag_relation (
        sag_registrering_id,
        virkning,
        rel_maal_uuid,
        rel_maal_urn,
        rel_type,
!       objekt_type,
!       rel_index,
!       rel_type_spec,
!       journal_notat,
!       journal_dokument_attr
      )
      SELECT
        sag_registrering_id,
        a.virkning,
        a.relMaalUuid,
        a.relMaalUrn,
        a.relType,
!       a.objektType,
!         CASE WHEN a.relType = any (sag_rel_type_cardinality_unlimited) THEN --rel_index
!         nextval('sag_rel_' || a.relType::text || sag_uuid_underscores)
!         ELSE 
!         NULL
!         END,
!         CASE 
!           WHEN a.relType='journalpost' THEN a.relTypeSpec  --rel_type_spec
!           ELSE
!           NULL
!         END,
!       a.journalNotat,
!       a.journalDokumentAttr
      FROM unnest(sag_registrering.relationer) a
!     WHERE 
!           a.relMaalUuid IS NOT NULL 
!       OR (a.relMaalUrn IS NOT NULL AND a.relMaalUrn<>'') 
!       OR (  
!           (NOT (a.journalNotat IS NULL)) 
!           AND ( (a.journalNotat).titel <>'' OR (a.journalNotat).notat <>'' OR (a.journalNotat).format<>'' ) 
!         )
    ;
  
+ 
+ --Drop temporary sequences
+ FOREACH sag_relation_kode IN ARRAY (SELECT array_agg( DISTINCT a.RelType) FROM  unnest(sag_registrering.relationer) a WHERE a.RelType = any (sag_rel_type_cardinality_unlimited))
+   LOOP
+   sag_rel_seq_name := 'sag_rel_' || sag_relation_kode::text || sag_uuid_underscores;
+   EXECUTE 'DROP  SEQUENCE ' || sag_rel_seq_name || ';';
+ END LOOP;
+ 
+ 
+ END IF;
+ 
    PERFORM amqp.publish(1, 'mox.notifications', '', format('create %s', sag_uuid));
  
  RETURN sag_uuid;
  
  END;
