*** ../generated-files/as_create_or_import_dokument.sql	2015-08-03 10:57:13.458770871 +0200
--- ../patches/as_create_or_import_dokument.org.sql	2015-08-03 10:55:44.798766455 +0200
***************
*** 20,30 ****
    dokument_attr_egenskaber_obj dokumentEgenskaberAttrType;
    
    dokument_tils_fremdrift_obj dokumentFremdriftTilsType;
    
    dokument_relationer DokumentRelationType;
! 
  BEGIN
  
  IF dokument_uuid IS NULL THEN
      LOOP
      dokument_uuid:=uuid_generate_v4();
--- 20,36 ----
    dokument_attr_egenskaber_obj dokumentEgenskaberAttrType;
    
    dokument_tils_fremdrift_obj dokumentFremdriftTilsType;
    
    dokument_relationer DokumentRelationType;
!   dokument_variant_obj DokumentVariantType;
!   dokument_variant_egenskab_obj DokumentVariantEgenskaberType;
!   dokument_del_obj DokumentDelType;
!   dokument_del_egenskaber_obj DokumentDelEgenskaberType;
!   dokument_del_relation_obj DokumentDelRelationType;
!   dokument_variant_new_id bigint;
!   dokument_del_new_id bigint;
  BEGIN
  
  IF dokument_uuid IS NULL THEN
      LOOP
      dokument_uuid:=uuid_generate_v4();
***************
*** 192,201 ****
--- 198,346 ----
      FROM unnest(dokument_registrering.relationer) a
      WHERE (a.relMaalUuid IS NOT NULL OR (a.relMaalUrn IS NOT NULL AND a.relMaalUrn<>'') )
    ;
  
  
+ /*********************************/
+ --Insert document variants (and parts)
+ 
+ IF dokument_registrering.varianter IS NOT NULL AND coalesce(array_length(dokument_registrering.varianter,1),0)>0 THEN
+   
+ 
+ FOREACH dokument_variant_obj IN ARRAY dokument_registrering.varianter
+ LOOP
+ 
+ dokument_variant_new_id:=nextval('dokument_variant_id_seq'::regclass);
+ 
+   INSERT INTO dokument_variant (
+       id,
+         varianttekst,
+           dokument_registrering_id
+   )
+   VALUES
+   (
+       dokument_variant_new_id,
+         dokument_variant_obj.varianttekst,
+           dokument_registrering_id
+   ); 
+ 
+ 
+   IF dokument_variant_obj.egenskaber IS NOT NULL AND coalesce(array_length(dokument_variant_obj.egenskaber,1),0)>0 THEN
+ 
+     FOREACH dokument_variant_egenskab_obj IN ARRAY dokument_variant_obj.egenskaber
+     LOOP
+ 
+      INSERT INTO dokument_variant_egenskaber(
+       variant_id,
+         varianttekst,
+           arkivering, 
+             delvisscannet, 
+               offentliggoerelse, 
+                 produktion,
+                   virkning
+       )
+       VALUES (
+       dokument_variant_new_id,  
+         dokument_variant_obj.varianttekst,
+           dokument_variant_obj_egenskab.arkivering,
+             dokument_variant_obj_egenskab.delvisscannet,
+               dokument_variant_obj_egenskab.offentliggoerelse,
+                 dokument_variant_obj_egenskab.produktion,
+                   dokument_variant_obj_egenskab.virkning
+         )
+     ;
+ 
+     END LOOP; --variant_egenskaber
+   END IF; --variant_egenskaber
+ 
+ 
+   IF dokument_variant_obj.dele IS NOT NULL AND coalesce(array_length(dokument_variant_obj.dele,1),0)>0 THEN
+ 
+     FOREACH dokument_del_obj IN ARRAY dokument_variant_obj.dele
+     LOOP
+ 
+     dokument_del_new_id:=nextval('dokument_del_id_seq'::regclass);
+ 
+   INSERT INTO dokument_del (
+     id,
+       deltekst,
+         variant_id
+     )
+     VALUES
+     (
+     dokument_del_new_id,
+         dokument_del_obj.deltekst,
+           dokument_variant_new_id
+     )
+     ;
+ 
+     IF dokument_del_obj.egenskaber IS NOT NULL AND coalesce(array_length(dokument_del_obj.egenskaber,1),0)>0 THEN
+ 
+     FOREACH dokument_del_egenskaber_obj IN ARRAY dokument_del_obj.egenskaber
+     LOOP
+ 
+     INSERT INTO
+     dokument_del_egenskaber (
+       del_id
+         indeks, 
+           indhold, 
+             lokation, 
+               mimetype, 
+                 virkning
+     )
+     VALUES
+     (
+       dokument_del_new_id, 
+         dokument_del_egenskaber_obj.indeks,
+           dokument_del_egenskaber_obj.indhold,
+             dokument_del_egenskaber_obj.lokation,
+               dokument_del_egenskaber_obj.mimetype,
+                 dokument_del_egenskaber_obj.virkning
+     )
+     ;                
+ 
+     END LOOP;--del_egenskaber
+     END IF; --del_egenskaber
+ 
+     IF dokument_del_obj.relationer IS NOT NULL AND coalesce(array_length(dokument_del_obj.relationer,1),0)>0 THEN
+ 
+     FOREACH dokument_del_relation_obj IN ARRAY dokument_del_obj.relationer
+     LOOP
+ 
+       INSERT INTO dokument_del_relation (
+         del_id,
+           virkning,
+             rel_maal_uuid, 
+               rel_maal_urn,
+                 rel_type,
+                   objekt_type
+       )
+       VALUES
+       (
+         dokument_del_new_id,
+           dokument_del_relation_obj.virkning,
+             dokument_del_relation_obj.relMaalUuid,
+               dokument_del_relation_obj.relMaalUrn,
+                 dokument_del_relation_obj.relType,
+                   dokument_del_relation_obj.objektType
+       )
+       ;
+ 
+     END LOOP;--del_relationer
+ 
+     END IF; --dokument_del_obj.relationer
+ 
+     END LOOP; --variant_dele
+   END IF; 
+ 
+  END LOOP; --varianter
+ 
+ 
+ END IF; --varianter
+ 
+ 
+ 
  RETURN dokument_uuid;
  
  END;
  $$ LANGUAGE plpgsql VOLATILE;
  
