*** ./generated-files/as_search_aktivitet.sql	2016-11-27 12:19:14.000000000 +0100
--- ./patches/as_search_aktivitet.org.sql	2016-11-27 12:15:53.000000000 +0100
***************
*** 776,785 ****
--- 776,821 ----
  					relationTypeObj.urn IS NULL
  					OR
  					relationTypeObj.urn = a.rel_maal_urn
  				)
  				AND
+ 				(
+  					relationTypeObj.indeks IS NULL
+  					OR
+  					relationTypeObj.indeks = a.rel_index
+  				)
+ 				AND
+ 				(
+ 					relationTypeObj.aktoerAttr IS NULL
+ 					OR
+ 					(
+ 						(
+ 							(relationTypeObj.aktoerAttr).obligatorik IS NULL
+ 							OR
+ 							(relationTypeObj.aktoerAttr).obligatorik = (a.aktoer_attr).obligatorik					
+ 						)
+ 						AND
+ 						(
+ 							(relationTypeObj.aktoerAttr).accepteret IS NULL
+ 							OR
+ 							(relationTypeObj.aktoerAttr).accepteret = (a.aktoer_attr).accepteret
+ 						)
+ 						AND
+ 						(
+ 							(relationTypeObj.aktoerAttr).repraesentation_uuid IS NULL
+ 							OR
+ 							(relationTypeObj.aktoerAttr).repraesentation_uuid = (a.aktoer_attr).repraesentation_uuid
+ 						)
+ 						AND
+ 						(
+ 							(relationTypeObj.aktoerAttr).repraesentation_urn IS NULL
+ 							OR
+ 							(relationTypeObj.aktoerAttr).repraesentation_urn = (a.aktoer_attr).repraesentation_urn
+ 						)
+ 					)
+ 				) 
+ 				AND
  						(
  				(registreringObj.registrering) IS NULL 
  				OR
  				(
  					(
***************
*** 864,874 ****
  			SELECT DISTINCT
  			b.aktivitet_id 
  			FROM  aktivitet_relation a
  			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
  			WHERE
! 			anyuuid = a.rel_maal_uuid
  			AND
  			(
  				virkningSoeg IS NULL
  				OR
  				virkningSoeg && (a.virkning).TimePeriod
--- 900,914 ----
  			SELECT DISTINCT
  			b.aktivitet_id 
  			FROM  aktivitet_relation a
  			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
  			WHERE
! 			(
! 				anyuuid = a.rel_maal_uuid
! 			OR  
! 				((NOT (a.aktoer_attr IS NULL)) AND anyuuid = (a.aktoer_attr).repraesentation_uuid )
! 			)
  			AND
  			(
  				virkningSoeg IS NULL
  				OR
  				virkningSoeg && (a.virkning).TimePeriod
***************
*** 959,969 ****
  			SELECT DISTINCT
  			b.aktivitet_id 
  			FROM  aktivitet_relation a
  			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
  			WHERE
! 			anyurn = a.rel_maal_urn
  			AND
  			(
  				virkningSoeg IS NULL
  				OR
  				virkningSoeg && (a.virkning).TimePeriod
--- 999,1013 ----
  			SELECT DISTINCT
  			b.aktivitet_id 
  			FROM  aktivitet_relation a
  			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
  			WHERE
! 			(
! 				anyurn = a.rel_maal_urn
! 				OR 
! 				((NOT (a.aktoer_attr IS NULL)) AND anyurn = (a.aktoer_attr).repraesentation_urn)
! 			)
  			AND
  			(
  				virkningSoeg IS NULL
  				OR
  				virkningSoeg && (a.virkning).TimePeriod
