*** ./generated-files/as_search_indsats.sql	2016-12-04 14:36:50.000000000 +0100
--- ./patches/as_search_indsats.org.sql	2016-12-04 14:33:48.000000000 +0100
***************
*** 3,15 ****
  --
  -- This Source Code Form is subject to the terms of the Mozilla Public
  -- License, v. 2.0. If a copy of the MPL was not distributed with this
  -- file, You can obtain one at http://mozilla.org/MPL/2.0/.
  
- /*
- NOTICE: This file is auto-generated using the script: apply-template.py indsats as_search.jinja.sql
- */
  
  
  CREATE OR REPLACE FUNCTION as_search_indsats(
  	firstResult int,--TOOD ??
  	indsats_uuid uuid,
--- 3,12 ----
***************
*** 17,27 ****
  	virkningSoeg TSTZRANGE, -- = TSTZRANGE(current_timestamp,current_timestamp,'[]'),
  	maxResults int = 2147483647,
  	anyAttrValueArr text[] = '{}'::text[],
  	anyuuidArr	uuid[] = '{}'::uuid[],
  	anyurnArr text[] = '{}'::text[],
! 	auth_criteria_arr IndsatsRegistreringType[]=null
  	)
    RETURNS uuid[] AS 
  $$
  DECLARE
  	indsats_candidates uuid[];
--- 14,26 ----
  	virkningSoeg TSTZRANGE, -- = TSTZRANGE(current_timestamp,current_timestamp,'[]'),
  	maxResults int = 2147483647,
  	anyAttrValueArr text[] = '{}'::text[],
  	anyuuidArr	uuid[] = '{}'::uuid[],
  	anyurnArr text[] = '{}'::text[],
! 	auth_criteria_arr IndsatsRegistreringType[]=null,
! 	search_operator_greater_than_or_equal_attr_egenskaber IndsatsEgenskaberAttrType[]=null,
!   	search_operator_less_than_or_equal_attr_egenskaber IndsatsEgenskaberAttrType[]=null
  	)
    RETURNS uuid[] AS 
  $$
  DECLARE
  	indsats_candidates uuid[];
***************
*** 755,764 ****
--- 754,769 ----
  					relationTypeObj.urn IS NULL
  					OR
  					relationTypeObj.urn = a.rel_maal_urn
  				)
  				AND
+ 				(
+  					relationTypeObj.indeks IS NULL
+  					OR
+  					relationTypeObj.indeks = a.rel_index
+  				)
+ 				AND
  						(
  				(registreringObj.registrering) IS NULL 
  				OR
  				(
  					(
***************
*** 1023,1034 ****
--- 1028,1356 ----
  	END LOOP;
  END IF;
  
  --/**********************//
  
+  --/**********************************************************//
+ --Filtration using operator 'greather than or equal': Egenskaber
+ --/**********************************************************//
+ IF coalesce(array_length(search_operator_greater_than_or_equal_attr_egenskaber,1),0)>0 THEN
+ 	IF (coalesce(array_length(indsats_candidates,1),0)>0 OR NOT indsats_candidates_is_initialized) THEN
+ 		FOREACH attrEgenskaberTypeObj IN ARRAY search_operator_greater_than_or_equal_attr_egenskaber
+ 		LOOP
+ 
+ indsats_candidates:=array(
+ 			SELECT DISTINCT
+ 			b.indsats_id 
+ 			FROM  indsats_attr_egenskaber a
+ 			JOIN indsats_registrering b on a.indsats_registrering_id=b.id
+ 			WHERE
+ 				(
+ 					(
+ 						attrEgenskaberTypeObj.virkning IS NULL 
+ 						OR
+ 						(
+ 							(
+ 								(
+ 							 		(attrEgenskaberTypeObj.virkning).TimePeriod IS NULL
+ 								)
+ 								OR
+ 								(
+ 									(attrEgenskaberTypeObj.virkning).TimePeriod && (a.virkning).TimePeriod
+ 								)
+ 							)
+ 							AND
+ 							(
+ 									(attrEgenskaberTypeObj.virkning).AktoerRef IS NULL OR (attrEgenskaberTypeObj.virkning).AktoerRef=(a.virkning).AktoerRef
+ 							)
+ 							AND
+ 							(
+ 									(attrEgenskaberTypeObj.virkning).AktoerTypeKode IS NULL OR (attrEgenskaberTypeObj.virkning).AktoerTypeKode=(a.virkning).AktoerTypeKode
+ 							)
+ 							AND
+ 							(
+ 									(attrEgenskaberTypeObj.virkning).NoteTekst IS NULL OR  (a.virkning).NoteTekst ILIKE (attrEgenskaberTypeObj.virkning).NoteTekst  
+ 							)
+ 						)
+ 					)
+ 				)
+ 				AND
+ 				(
+ 					(NOT (attrEgenskaberTypeObj.virkning IS NULL OR (attrEgenskaberTypeObj.virkning).TimePeriod IS NULL)) --we have already filtered on virkning above
+ 					OR
+ 					(
+ 						virkningSoeg IS NULL
+ 						OR
+ 						virkningSoeg && (a.virkning).TimePeriod
+ 					)
+ 				)
+ 				AND
+ 				(
+ 					attrEgenskaberTypeObj.brugervendtnoegle IS NULL
+ 					OR 
+ 					a.brugervendtnoegle >= attrEgenskaberTypeObj.brugervendtnoegle  
+ 				)
+ 				AND
+ 				(
+ 					attrEgenskaberTypeObj.beskrivelse IS NULL
+ 					OR 
+ 					a.beskrivelse >= attrEgenskaberTypeObj.beskrivelse 
+ 				)
+ 				AND
+ 				(
+ 					attrEgenskaberTypeObj.starttidspunkt IS NULL
+ 					OR 
+ 					a.starttidspunkt >= attrEgenskaberTypeObj.starttidspunkt 
+ 				)
+ 				AND
+ 				(
+ 					attrEgenskaberTypeObj.sluttidspunkt IS NULL
+ 					OR 
+ 					a.sluttidspunkt >= attrEgenskaberTypeObj.sluttidspunkt 
+ 				)
+ 				AND
+ 						(
+ 				(registreringObj.registrering) IS NULL 
+ 				OR
+ 				(
+ 					(
+ 						(registreringObj.registrering).timeperiod IS NULL 
+ 						OR
+ 						(registreringObj.registrering).timeperiod && (b.registrering).timeperiod
+ 					)
+ 					AND
+ 					(
+ 						(registreringObj.registrering).livscykluskode IS NULL 
+ 						OR
+ 						(registreringObj.registrering).livscykluskode = (b.registrering).livscykluskode 		
+ 					) 
+ 					AND
+ 					(
+ 						(registreringObj.registrering).brugerref IS NULL
+ 						OR
+ 						(registreringObj.registrering).brugerref = (b.registrering).brugerref
+ 					)
+ 					AND
+ 					(
+ 						(registreringObj.registrering).note IS NULL
+ 						OR
+ 						(b.registrering).note ILIKE (registreringObj.registrering).note
+ 					)
+ 			)
+ 		)
+ 		AND
+ 		(
+ 			(
+ 				((b.registrering).livscykluskode <> 'Slettet'::Livscykluskode )
+ 				AND
+ 					(
+ 						(registreringObj.registrering) IS NULL 
+ 						OR
+ 						(registreringObj.registrering).livscykluskode IS NULL 
+ 					)
+ 			)
+ 			OR
+ 			(
+ 				(NOT ((registreringObj.registrering) IS NULL))
+ 				AND
+ 				(registreringObj.registrering).livscykluskode IS NOT NULL 
+ 			)
+ 		)
+ 		AND
+ 		(
+ 			(
+ 			  (
+ 			  	(registreringObj.registrering) IS NULL
+ 			  	OR
+ 			  	(registreringObj.registrering).timeperiod IS NULL
+ 			  )
+ 			  AND
+ 			  upper((b.registrering).timeperiod)='infinity'::TIMESTAMPTZ
+ 			)  	
+ 		OR
+ 			(
+ 				(NOT ((registreringObj.registrering) IS NULL))
+ 				AND
+ 				((registreringObj.registrering).timeperiod IS NOT NULL)
+ 			)
+ 		)
+ 		AND
+ 		( (NOT indsats_candidates_is_initialized) OR b.indsats_id = ANY (indsats_candidates) )
+ 
+ 			);
+ 			
+ 
+ 			indsats_candidates_is_initialized:=true;
+ 			
+ 
+ 		END LOOP;
+ 	END IF;
+ END IF;
+ 
   
  
+ --/**********************//
+ 
+  --/**********************************************************//
+ --Filtration using operator 'less than or equal': Egenskaber
+ --/**********************************************************//
+ IF coalesce(array_length(search_operator_less_than_or_equal_attr_egenskaber,1),0)>0 THEN
+ 	IF (coalesce(array_length(indsats_candidates,1),0)>0 OR NOT indsats_candidates_is_initialized) THEN
+ 		FOREACH attrEgenskaberTypeObj IN ARRAY search_operator_less_than_or_equal_attr_egenskaber
+ 		LOOP
+ 
+ indsats_candidates:=array(
+ 			SELECT DISTINCT
+ 			b.indsats_id 
+ 			FROM  indsats_attr_egenskaber a
+ 			JOIN indsats_registrering b on a.indsats_registrering_id=b.id
+ 			WHERE
+ 				(
+ 					(
+ 						attrEgenskaberTypeObj.virkning IS NULL 
+ 						OR
+ 						(
+ 							(
+ 								(
+ 							 		(attrEgenskaberTypeObj.virkning).TimePeriod IS NULL
+ 								)
+ 								OR
+ 								(
+ 									(attrEgenskaberTypeObj.virkning).TimePeriod && (a.virkning).TimePeriod
+ 								)
+ 							)
+ 							AND
+ 							(
+ 									(attrEgenskaberTypeObj.virkning).AktoerRef IS NULL OR (attrEgenskaberTypeObj.virkning).AktoerRef=(a.virkning).AktoerRef
+ 							)
+ 							AND
+ 							(
+ 									(attrEgenskaberTypeObj.virkning).AktoerTypeKode IS NULL OR (attrEgenskaberTypeObj.virkning).AktoerTypeKode=(a.virkning).AktoerTypeKode
+ 							)
+ 							AND
+ 							(
+ 									(attrEgenskaberTypeObj.virkning).NoteTekst IS NULL OR  (a.virkning).NoteTekst ILIKE (attrEgenskaberTypeObj.virkning).NoteTekst  
+ 							)
+ 						)
+ 					)
+ 				)
+ 				AND
+ 				(
+ 					(NOT (attrEgenskaberTypeObj.virkning IS NULL OR (attrEgenskaberTypeObj.virkning).TimePeriod IS NULL)) --we have already filtered on virkning above
+ 					OR
+ 					(
+ 						virkningSoeg IS NULL
+ 						OR
+ 						virkningSoeg && (a.virkning).TimePeriod
+ 					)
+ 				)
+ 				AND
+ 				(
+ 					attrEgenskaberTypeObj.brugervendtnoegle IS NULL
+ 					OR 
+ 					a.brugervendtnoegle <= attrEgenskaberTypeObj.brugervendtnoegle  
+ 				)
+ 				AND
+ 				(
+ 					attrEgenskaberTypeObj.beskrivelse IS NULL
+ 					OR 
+ 					a.beskrivelse <= attrEgenskaberTypeObj.beskrivelse 
+ 				)
+ 				AND
+ 				(
+ 					attrEgenskaberTypeObj.starttidspunkt IS NULL
+ 					OR 
+ 					a.starttidspunkt <= attrEgenskaberTypeObj.starttidspunkt 
+ 				)
+ 				AND
+ 				(
+ 					attrEgenskaberTypeObj.sluttidspunkt IS NULL
+ 					OR 
+ 					a.sluttidspunkt <= attrEgenskaberTypeObj.sluttidspunkt 
+ 				)
+ 				AND
+ 						(
+ 				(registreringObj.registrering) IS NULL 
+ 				OR
+ 				(
+ 					(
+ 						(registreringObj.registrering).timeperiod IS NULL 
+ 						OR
+ 						(registreringObj.registrering).timeperiod && (b.registrering).timeperiod
+ 					)
+ 					AND
+ 					(
+ 						(registreringObj.registrering).livscykluskode IS NULL 
+ 						OR
+ 						(registreringObj.registrering).livscykluskode = (b.registrering).livscykluskode 		
+ 					) 
+ 					AND
+ 					(
+ 						(registreringObj.registrering).brugerref IS NULL
+ 						OR
+ 						(registreringObj.registrering).brugerref = (b.registrering).brugerref
+ 					)
+ 					AND
+ 					(
+ 						(registreringObj.registrering).note IS NULL
+ 						OR
+ 						(b.registrering).note ILIKE (registreringObj.registrering).note
+ 					)
+ 			)
+ 		)
+ 		AND
+ 		(
+ 			(
+ 				((b.registrering).livscykluskode <> 'Slettet'::Livscykluskode )
+ 				AND
+ 					(
+ 						(registreringObj.registrering) IS NULL 
+ 						OR
+ 						(registreringObj.registrering).livscykluskode IS NULL 
+ 					)
+ 			)
+ 			OR
+ 			(
+ 				(NOT ((registreringObj.registrering) IS NULL))
+ 				AND
+ 				(registreringObj.registrering).livscykluskode IS NOT NULL 
+ 			)
+ 		)
+ 		AND
+ 		(
+ 			(
+ 			  (
+ 			  	(registreringObj.registrering) IS NULL
+ 			  	OR
+ 			  	(registreringObj.registrering).timeperiod IS NULL
+ 			  )
+ 			  AND
+ 			  upper((b.registrering).timeperiod)='infinity'::TIMESTAMPTZ
+ 			)  	
+ 		OR
+ 			(
+ 				(NOT ((registreringObj.registrering) IS NULL))
+ 				AND
+ 				((registreringObj.registrering).timeperiod IS NOT NULL)
+ 			)
+ 		)
+ 		AND
+ 		( (NOT indsats_candidates_is_initialized) OR b.indsats_id = ANY (indsats_candidates) )
+ 
+ 			);
+ 			
+ 
+ 			indsats_candidates_is_initialized:=true;
+ 			
+ 
+ 		END LOOP;
+ 	END IF;
+ END IF;
+ 
+  
+ 
+ --/**********************//
  
  
  --RAISE DEBUG 'indsats_candidates_is_initialized step 5:%',indsats_candidates_is_initialized;
  --RAISE DEBUG 'indsats_candidates step 5:%',indsats_candidates;
  
