*** ./generated-files/as_search_aktivitet.sql	2016-11-27 19:55:05.000000000 +0100
--- ./patches/as_search_aktivitet.org.sql	2016-11-27 19:42:35.000000000 +0100
***************
*** 17,27 ****
  	virkningSoeg TSTZRANGE, -- = TSTZRANGE(current_timestamp,current_timestamp,'[]'),
  	maxResults int = 2147483647,
  	anyAttrValueArr text[] = '{}'::text[],
  	anyuuidArr	uuid[] = '{}'::uuid[],
  	anyurnArr text[] = '{}'::text[],
! 	auth_criteria_arr AktivitetRegistreringType[]=null
  	)
    RETURNS uuid[] AS 
  $$
  DECLARE
  	aktivitet_candidates uuid[];
--- 17,29 ----
  	virkningSoeg TSTZRANGE, -- = TSTZRANGE(current_timestamp,current_timestamp,'[]'),
  	maxResults int = 2147483647,
  	anyAttrValueArr text[] = '{}'::text[],
  	anyuuidArr	uuid[] = '{}'::uuid[],
  	anyurnArr text[] = '{}'::text[],
! 	auth_criteria_arr AktivitetRegistreringType[]=null,
! 	search_operator_greater_than_or_equal_attr_egenskaber AktivitetEgenskaberAttrType[]=null,
! 	search_operator_less_than_or_equal_attr_egenskaber AktivitetEgenskaberAttrType[]=null
  	)
    RETURNS uuid[] AS 
  $$
  DECLARE
  	aktivitet_candidates uuid[];
***************
*** 776,785 ****
--- 778,823 ----
  					relationTypeObj.urn IS NULL
  					OR
  					relationTypeObj.urn = a.rel_maal_urn
  				)
  				AND
+ 				(
+  					relationTypeObj.indeks IS NULL
+  					OR
+  					relationTypeObj.indeks = a.rel_index
+  				)
+ 				AND
+ 				(
+ 					relationTypeObj.aktoerAttr IS NULL
+ 					OR
+ 					(
+ 						(
+ 							(relationTypeObj.aktoerAttr).obligatorisk IS NULL
+ 							OR
+ 							(relationTypeObj.aktoerAttr).obligatorisk = (a.aktoer_attr).obligatorisk					
+ 						)
+ 						AND
+ 						(
+ 							(relationTypeObj.aktoerAttr).accepteret IS NULL
+ 							OR
+ 							(relationTypeObj.aktoerAttr).accepteret = (a.aktoer_attr).accepteret
+ 						)
+ 						AND
+ 						(
+ 							(relationTypeObj.aktoerAttr).repraesentation_uuid IS NULL
+ 							OR
+ 							(relationTypeObj.aktoerAttr).repraesentation_uuid = (a.aktoer_attr).repraesentation_uuid
+ 						)
+ 						AND
+ 						(
+ 							(relationTypeObj.aktoerAttr).repraesentation_urn IS NULL
+ 							OR
+ 							(relationTypeObj.aktoerAttr).repraesentation_urn = (a.aktoer_attr).repraesentation_urn
+ 						)
+ 					)
+ 				) 
+ 				AND
  						(
  				(registreringObj.registrering) IS NULL 
  				OR
  				(
  					(
***************
*** 864,874 ****
  			SELECT DISTINCT
  			b.aktivitet_id 
  			FROM  aktivitet_relation a
  			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
  			WHERE
! 			anyuuid = a.rel_maal_uuid
  			AND
  			(
  				virkningSoeg IS NULL
  				OR
  				virkningSoeg && (a.virkning).TimePeriod
--- 902,916 ----
  			SELECT DISTINCT
  			b.aktivitet_id 
  			FROM  aktivitet_relation a
  			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
  			WHERE
! 			(
! 				anyuuid = a.rel_maal_uuid
! 			OR  
! 				((NOT (a.aktoer_attr IS NULL)) AND anyuuid = (a.aktoer_attr).repraesentation_uuid )
! 			)
  			AND
  			(
  				virkningSoeg IS NULL
  				OR
  				virkningSoeg && (a.virkning).TimePeriod
***************
*** 959,969 ****
  			SELECT DISTINCT
  			b.aktivitet_id 
  			FROM  aktivitet_relation a
  			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
  			WHERE
! 			anyurn = a.rel_maal_urn
  			AND
  			(
  				virkningSoeg IS NULL
  				OR
  				virkningSoeg && (a.virkning).TimePeriod
--- 1001,1015 ----
  			SELECT DISTINCT
  			b.aktivitet_id 
  			FROM  aktivitet_relation a
  			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
  			WHERE
! 			(
! 				anyurn = a.rel_maal_urn
! 				OR 
! 				((NOT (a.aktoer_attr IS NULL)) AND anyurn = (a.aktoer_attr).repraesentation_urn)
! 			)
  			AND
  			(
  				virkningSoeg IS NULL
  				OR
  				virkningSoeg && (a.virkning).TimePeriod
***************
*** 1044,1054 ****
  	END LOOP;
  END IF;
  
  --/**********************//
  
!  
  
  
  
  --RAISE DEBUG 'aktivitet_candidates_is_initialized step 5:%',aktivitet_candidates_is_initialized;
  --RAISE DEBUG 'aktivitet_candidates step 5:%',aktivitet_candidates;
--- 1090,1452 ----
  	END LOOP;
  END IF;
  
  --/**********************//
  
!  --/**********************************************************//
! --Filtration using operator 'greather than or equal': Egenskaber
! --/**********************************************************//
! IF coalesce(array_length(search_operator_greater_than_or_equal_attr_egenskaber,1),0)>0 THEN
! 	IF (coalesce(array_length(aktivitet_candidates,1),0)>0 OR NOT aktivitet_candidates_is_initialized) THEN
! 		FOREACH attrEgenskaberTypeObj IN ARRAY search_operator_greater_than_or_equal_attr_egenskaber
! 		LOOP
! 			aktivitet_candidates:=array(
! 			SELECT DISTINCT
! 			b.aktivitet_id 
! 			FROM  aktivitet_attr_egenskaber a
! 			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
! 			WHERE
! 				(
! 					(
! 						attrEgenskaberTypeObj.virkning IS NULL 
! 						OR
! 						(
! 							(
! 								(
! 							 		(attrEgenskaberTypeObj.virkning).TimePeriod IS NULL
! 								)
! 								OR
! 								(
! 									(attrEgenskaberTypeObj.virkning).TimePeriod && (a.virkning).TimePeriod
! 								)
! 							)
! 							AND
! 							(
! 									(attrEgenskaberTypeObj.virkning).AktoerRef IS NULL OR (attrEgenskaberTypeObj.virkning).AktoerRef=(a.virkning).AktoerRef
! 							)
! 							AND
! 							(
! 									(attrEgenskaberTypeObj.virkning).AktoerTypeKode IS NULL OR (attrEgenskaberTypeObj.virkning).AktoerTypeKode=(a.virkning).AktoerTypeKode
! 							)
! 							AND
! 							(
! 									(attrEgenskaberTypeObj.virkning).NoteTekst IS NULL OR  (a.virkning).NoteTekst ILIKE (attrEgenskaberTypeObj.virkning).NoteTekst  
! 							)
! 						)
! 					)
! 				)
! 				AND
! 				(
! 					(NOT (attrEgenskaberTypeObj.virkning IS NULL OR (attrEgenskaberTypeObj.virkning).TimePeriod IS NULL)) --we have already filtered on virkning above
! 					OR
! 					(
! 						virkningSoeg IS NULL
! 						OR
! 						virkningSoeg && (a.virkning).TimePeriod
! 					)
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.brugervendtnoegle IS NULL
! 					OR 
! 					a.brugervendtnoegle >= attrEgenskaberTypeObj.brugervendtnoegle 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.aktivitetnavn IS NULL
! 					OR 
! 					a.aktivitetnavn >= attrEgenskaberTypeObj.aktivitetnavn 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.beskrivelse IS NULL
! 					OR 
! 					a.beskrivelse >= attrEgenskaberTypeObj.beskrivelse 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.starttidspunkt IS NULL
! 					OR 
! 					a.starttidspunkt >= attrEgenskaberTypeObj.starttidspunkt 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.sluttidspunkt IS NULL
! 					OR 
! 					a.sluttidspunkt >= attrEgenskaberTypeObj.sluttidspunkt 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.tidsforbrug IS NULL
! 					OR 
! 					a.tidsforbrug >= attrEgenskaberTypeObj.tidsforbrug 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.formaal IS NULL
! 					OR 
! 					a.formaal >= attrEgenskaberTypeObj.formaal 
! 				)
! 				AND
! 						(
! 				(registreringObj.registrering) IS NULL 
! 				OR
! 				(
! 					(
! 						(registreringObj.registrering).timeperiod IS NULL 
! 						OR
! 						(registreringObj.registrering).timeperiod && (b.registrering).timeperiod
! 					)
! 					AND
! 					(
! 						(registreringObj.registrering).livscykluskode IS NULL 
! 						OR
! 						(registreringObj.registrering).livscykluskode = (b.registrering).livscykluskode 		
! 					) 
! 					AND
! 					(
! 						(registreringObj.registrering).brugerref IS NULL
! 						OR
! 						(registreringObj.registrering).brugerref = (b.registrering).brugerref
! 					)
! 					AND
! 					(
! 						(registreringObj.registrering).note IS NULL
! 						OR
! 						(b.registrering).note ILIKE (registreringObj.registrering).note
! 					)
! 			)
! 		)
! 		AND
! 		(
! 			(
! 				((b.registrering).livscykluskode <> 'Slettet'::Livscykluskode )
! 				AND
! 					(
! 						(registreringObj.registrering) IS NULL 
! 						OR
! 						(registreringObj.registrering).livscykluskode IS NULL 
! 					)
! 			)
! 			OR
! 			(
! 				(NOT ((registreringObj.registrering) IS NULL))
! 				AND
! 				(registreringObj.registrering).livscykluskode IS NOT NULL 
! 			)
! 		)
! 		AND
! 		(
! 			(
! 			  (
! 			  	(registreringObj.registrering) IS NULL
! 			  	OR
! 			  	(registreringObj.registrering).timeperiod IS NULL
! 			  )
! 			  AND
! 			  upper((b.registrering).timeperiod)='infinity'::TIMESTAMPTZ
! 			)  	
! 		OR
! 			(
! 				(NOT ((registreringObj.registrering) IS NULL))
! 				AND
! 				((registreringObj.registrering).timeperiod IS NOT NULL)
! 			)
! 		)
! 		AND
! 		( (NOT aktivitet_candidates_is_initialized) OR b.aktivitet_id = ANY (aktivitet_candidates) )
! 
! 			);
! 			
! 
! 			aktivitet_candidates_is_initialized:=true;
! 			
! 			
! 			END LOOP;
! 		END IF;	
! 	END IF;
! 
! --RAISE DEBUG 'aktivitet_candidates_is_initialized step 3:%',aktivitet_candidates_is_initialized;
! --RAISE DEBUG 'aktivitet_candidates step 3:%',aktivitet_candidates;
! 
!  --/**********************************************************//
! --Filtration using operator 'less than or equal': Egenskaber
! --/**********************************************************//
! IF coalesce(array_length(search_operator_less_than_or_equal_attr_egenskaber,1),0)>0 THEN
! 	IF (coalesce(array_length(aktivitet_candidates,1),0)>0 OR NOT aktivitet_candidates_is_initialized) THEN
! 		FOREACH attrEgenskaberTypeObj IN ARRAY search_operator_less_than_or_equal_attr_egenskaber
! 		LOOP
! 			aktivitet_candidates:=array(
! 			SELECT DISTINCT
! 			b.aktivitet_id 
! 			FROM  aktivitet_attr_egenskaber a
! 			JOIN aktivitet_registrering b on a.aktivitet_registrering_id=b.id
! 			WHERE
! 				(
! 					(
! 						attrEgenskaberTypeObj.virkning IS NULL 
! 						OR
! 						(
! 							(
! 								(
! 							 		(attrEgenskaberTypeObj.virkning).TimePeriod IS NULL
! 								)
! 								OR
! 								(
! 									(attrEgenskaberTypeObj.virkning).TimePeriod && (a.virkning).TimePeriod
! 								)
! 							)
! 							AND
! 							(
! 									(attrEgenskaberTypeObj.virkning).AktoerRef IS NULL OR (attrEgenskaberTypeObj.virkning).AktoerRef=(a.virkning).AktoerRef
! 							)
! 							AND
! 							(
! 									(attrEgenskaberTypeObj.virkning).AktoerTypeKode IS NULL OR (attrEgenskaberTypeObj.virkning).AktoerTypeKode=(a.virkning).AktoerTypeKode
! 							)
! 							AND
! 							(
! 									(attrEgenskaberTypeObj.virkning).NoteTekst IS NULL OR  (a.virkning).NoteTekst ILIKE (attrEgenskaberTypeObj.virkning).NoteTekst  
! 							)
! 						)
! 					)
! 				)
! 				AND
! 				(
! 					(NOT (attrEgenskaberTypeObj.virkning IS NULL OR (attrEgenskaberTypeObj.virkning).TimePeriod IS NULL)) --we have already filtered on virkning above
! 					OR
! 					(
! 						virkningSoeg IS NULL
! 						OR
! 						virkningSoeg && (a.virkning).TimePeriod
! 					)
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.brugervendtnoegle IS NULL
! 					OR 
! 					a.brugervendtnoegle <= attrEgenskaberTypeObj.brugervendtnoegle 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.aktivitetnavn IS NULL
! 					OR 
! 					a.aktivitetnavn <= attrEgenskaberTypeObj.aktivitetnavn 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.beskrivelse IS NULL
! 					OR 
! 					a.beskrivelse <= attrEgenskaberTypeObj.beskrivelse 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.starttidspunkt IS NULL
! 					OR 
! 					a.starttidspunkt <= attrEgenskaberTypeObj.starttidspunkt 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.sluttidspunkt IS NULL
! 					OR 
! 					a.sluttidspunkt <= attrEgenskaberTypeObj.sluttidspunkt 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.tidsforbrug IS NULL
! 					OR 
! 					a.tidsforbrug <= attrEgenskaberTypeObj.tidsforbrug 
! 				)
! 				AND
! 				(
! 					attrEgenskaberTypeObj.formaal IS NULL
! 					OR 
! 					a.formaal <= attrEgenskaberTypeObj.formaal 
! 				)
! 				AND
! 						(
! 				(registreringObj.registrering) IS NULL 
! 				OR
! 				(
! 					(
! 						(registreringObj.registrering).timeperiod IS NULL 
! 						OR
! 						(registreringObj.registrering).timeperiod && (b.registrering).timeperiod
! 					)
! 					AND
! 					(
! 						(registreringObj.registrering).livscykluskode IS NULL 
! 						OR
! 						(registreringObj.registrering).livscykluskode = (b.registrering).livscykluskode 		
! 					) 
! 					AND
! 					(
! 						(registreringObj.registrering).brugerref IS NULL
! 						OR
! 						(registreringObj.registrering).brugerref = (b.registrering).brugerref
! 					)
! 					AND
! 					(
! 						(registreringObj.registrering).note IS NULL
! 						OR
! 						(b.registrering).note ILIKE (registreringObj.registrering).note
! 					)
! 			)
! 		)
! 		AND
! 		(
! 			(
! 				((b.registrering).livscykluskode <> 'Slettet'::Livscykluskode )
! 				AND
! 					(
! 						(registreringObj.registrering) IS NULL 
! 						OR
! 						(registreringObj.registrering).livscykluskode IS NULL 
! 					)
! 			)
! 			OR
! 			(
! 				(NOT ((registreringObj.registrering) IS NULL))
! 				AND
! 				(registreringObj.registrering).livscykluskode IS NOT NULL 
! 			)
! 		)
! 		AND
! 		(
! 			(
! 			  (
! 			  	(registreringObj.registrering) IS NULL
! 			  	OR
! 			  	(registreringObj.registrering).timeperiod IS NULL
! 			  )
! 			  AND
! 			  upper((b.registrering).timeperiod)='infinity'::TIMESTAMPTZ
! 			)  	
! 		OR
! 			(
! 				(NOT ((registreringObj.registrering) IS NULL))
! 				AND
! 				((registreringObj.registrering).timeperiod IS NOT NULL)
! 			)
! 		)
! 		AND
! 		( (NOT aktivitet_candidates_is_initialized) OR b.aktivitet_id = ANY (aktivitet_candidates) )
! 
! 			);
! 			
! 
! 			aktivitet_candidates_is_initialized:=true;
! 			
! 			
! 			END LOOP;
! 		END IF;	
! 	END IF;
! 
! --RAISE DEBUG 'aktivitet_candidates_is_initialized step 3:%',aktivitet_candidates_is_initialized;
! --RAISE DEBUG 'aktivitet_candidates step 3:%',aktivitet_candidates;
! 
! --/**********************//
  
  
  
  --RAISE DEBUG 'aktivitet_candidates_is_initialized step 5:%',aktivitet_candidates_is_initialized;
  --RAISE DEBUG 'aktivitet_candidates step 5:%',aktivitet_candidates;
