*** ../generated-files/as_search_klasse.sql	2015-06-22 15:46:23.170794003 +0200
--- ../patches/as_search_klasse.sql.org	2015-06-22 16:01:40.450839686 +0200
***************
*** 31,40 ****
--- 31,42 ----
    	tilsPubliceretTypeObj KlassePubliceretTilsType;
  	relationTypeObj KlasseRelationType;
  	anyAttrValue text;
  	anyRelUuid uuid;
  	anyRelUrn text;
+ 	manipulatedAttrEgenskaberArr KlasseEgenskaberAttrType[]:='{}';
+ 	soegeordObj KlasseSoegeordType;
  BEGIN
  
  --RAISE DEBUG 'step 0:registreringObj:%',registreringObj;
  
  klasse_candidates_is_initialized := false;
***************
*** 140,162 ****
  --Filtration on attribute: Egenskaber
  --/**********************************************************//
  IF registreringObj IS NULL OR (registreringObj).attrEgenskaber IS NULL THEN
  	--RAISE DEBUG 'as_search_klasse: skipping filtration on attrEgenskaber';
  ELSE
  	IF (coalesce(array_length(klasse_candidates,1),0)>0 OR NOT klasse_candidates_is_initialized) THEN
! 		FOREACH attrEgenskaberTypeObj IN ARRAY registreringObj.attrEgenskaber
  		LOOP
  			klasse_candidates:=array(
  			SELECT DISTINCT
  			b.klasse_id 
  			FROM  klasse_attr_egenskaber a
  			JOIN klasse_registrering b on a.klasse_registrering_id=b.id
  			WHERE
  				(
  					(
- 						attrEgenskaberTypeObj.virkning IS NULL 
- 						OR
  						(
  							(
  								(
  							 		(attrEgenskaberTypeObj.virkning).TimePeriod IS NULL
  								)
--- 142,194 ----
  --Filtration on attribute: Egenskaber
  --/**********************************************************//
  IF registreringObj IS NULL OR (registreringObj).attrEgenskaber IS NULL THEN
  	--RAISE DEBUG 'as_search_klasse: skipping filtration on attrEgenskaber';
  ELSE
+ 
+ --To help facilitate the comparrison efforts (while diverging at a minimum form the templated db-kode, 
+ --we'll manipulate the attrEgenskaber array so to make sure that every object only has 1 sogeord element - duplicating the parent elements in attrEgenskaber as needed  )
+ 
+ FOREACH attrEgenskaberTypeObj IN ARRAY registreringObj.attrEgenskaber
+ LOOP
+ 	IF  (attrEgenskaberTypeObj).soegeord IS NULL OR coalesce(array_length((attrEgenskaberTypeObj).soegeord,1),0)<2 THEN
+ 	manipulatedAttrEgenskaberArr:=array_append(manipulatedAttrEgenskaberArr,attrEgenskaberTypeObj); --The element only has 0 or 1 soegeord element, så no manipulations is needed.
+ 	ELSE
+ 		FOREACH soegeordObj IN ARRAY (attrEgenskaberTypeObj).soegeord
+ 		LOOP
+ 			manipulatedAttrEgenskaberArr:=array_append(manipulatedAttrEgenskaberArr,
+ 				ROW (
+ 					attrEgenskaberTypeObj.brugervendtnoegle,
+ 					attrEgenskaberTypeObj.beskrivelse,
+ 					attrEgenskaberTypeObj.eksempel,
+ 					attrEgenskaberTypeObj.omfang,
+ 					attrEgenskaberTypeObj.titel,
+ 					attrEgenskaberTypeObj.retskilde,
+ 					attrEgenskaberTypeObj.aendringsnotat,
+ 					ARRAY[soegeordObj]::KlasseSoegeordType[], --NOTICE: Only 1 element in array
+ 					attrEgenskaberTypeObj.virkning
+ 					)::KlasseEgenskaberAttrType
+ 				);
+ 		END LOOP;
+ 	END IF;
+ END LOOP;
+ 
+ 
  	IF (coalesce(array_length(klasse_candidates,1),0)>0 OR NOT klasse_candidates_is_initialized) THEN
! 		FOREACH attrEgenskaberTypeObj IN ARRAY manipulatedAttrEgenskaberArr
  		LOOP
  			klasse_candidates:=array(
  			SELECT DISTINCT
  			b.klasse_id 
  			FROM  klasse_attr_egenskaber a
  			JOIN klasse_registrering b on a.klasse_registrering_id=b.id
+ 			LEFT JOIN klasse_attr_egenskaber_soegeord c on a.id=c.klasse_attr_egenskaber_id
  			WHERE
  				(
+ 					attrEgenskaberTypeObj.virkning IS NULL
+ 					OR
  					(
  						(
  							(
  								(
  							 		(attrEgenskaberTypeObj.virkning).TimePeriod IS NULL
  								)
***************
*** 231,241 ****
--- 263,297 ----
  					attrEgenskaberTypeObj.aendringsnotat IS NULL
  					OR 
  					a.aendringsnotat ILIKE attrEgenskaberTypeObj.aendringsnotat --case insensitive 
  				)
  				AND
+ 				(
+ 					(attrEgenskaberTypeObj.soegeord IS NULL OR array_length(attrEgenskaberTypeObj.soegeord,1)=0)
+ 					OR
+ 					(
+ 						(
+ 							(attrEgenskaberTypeObj.soegeord[1]).soegeordidentifikator IS NULL
+ 							OR
+ 							c.soegeordidentifikator ILIKE (attrEgenskaberTypeObj.soegeord[1]).soegeordidentifikator  
+ 						)
+ 						AND
  						(
+ 							(attrEgenskaberTypeObj.soegeord[1]).beskrivelse IS NULL
+ 							OR
+ 							c.beskrivelse ILIKE (attrEgenskaberTypeObj.soegeord[1]).beskrivelse  
+ 						)		
+ 						AND
+ 						(
+ 							(attrEgenskaberTypeObj.soegeord[1]).soegeordskategori IS NULL
+ 							OR
+ 							c.soegeordskategori ILIKE (attrEgenskaberTypeObj.soegeord[1]).soegeordskategori  
+ 						)
+ 					)
+ 				)
+ 				AND
+ 				(
  				(registreringObj.registrering) IS NULL 
  				OR
  				(
  					(
  						(registreringObj.registrering).timeperiod IS NULL 
***************
*** 324,333 ****
--- 380,390 ----
  
  			SELECT DISTINCT
  			b.klasse_id 
  			FROM  klasse_attr_egenskaber a
  			JOIN klasse_registrering b on a.klasse_registrering_id=b.id
+ 			LEFT JOIN klasse_attr_egenskaber_soegeord c on a.id=c.klasse_attr_egenskaber_id
  			WHERE
  			(
  				a.brugervendtnoegle ILIKE anyAttrValue
  				OR
  				a.beskrivelse ILIKE anyAttrValue
***************
*** 339,348 ****
--- 396,411 ----
  				a.titel ILIKE anyAttrValue
  				OR
  				a.retskilde ILIKE anyAttrValue
  				OR
  				a.aendringsnotat ILIKE anyAttrValue
+ 				OR 
+ 				c.soegeordidentifikator ILIKE anyAttrValue
+ 				OR 
+ 				c.beskrivelse ILIKE anyAttrValue
+ 				OR
+ 				c.soegeordskategori ILIKE anyAttrValue
  			)
  			AND
  			(
  				virkningSoeg IS NULL
  				OR
