*** ../generated-files/as_search_sag.sql	2015-08-14 13:42:17.857698758 +0200
--- as_search_sag.org.sql	2015-08-14 15:26:46.838010972 +0200
***************
*** 676,685 ****
--- 676,751 ----
  					relationTypeObj.urn IS NULL
  					OR
  					relationTypeObj.urn = a.rel_maal_urn
  				)
  				AND
+ 				(
+ 					relationTypeObj.indeks IS NULL
+ 					OR
+ 					relationTypeObj.indeks = a.rel_index
+ 				)
+ 				AND
+ 				(
+ 					relationTypeObj.relTypeSpec IS NULL
+ 					OR
+ 					relationTypeObj.relTypeSpec = rel_type_spec
+ 				)
+ 				AND
+ 				(
+ 					relationTypeObj.journalNotat IS NULL
+ 					OR
+ 					(
+ 						(
+ 							(relationTypeObj.journalNotat).titel IS NULL
+ 							OR
+ 						 	(a.journal_notat).titel ILIKE (relationTypeObj.journalNotat).titel
+ 						)
+ 						AND
+ 						(
+ 							(relationTypeObj.journalNotat).notat IS NULL
+ 							OR
+ 						 	(a.journal_notat).notat ILIKE (relationTypeObj.journalNotat).notat
+ 						)
+ 						AND
+ 						(
+ 							(relationTypeObj.journalNotat).format IS NULL
+ 							OR
+ 						 	(a.journal_notat).format ILIKE (relationTypeObj.journalNotat).format
+ 						)
+ 					)
+ 				)
+ 				AND
+ 				(
+ 					relationTypeObj.journalDokumentAttr IS NULL
+ 					OR
+ 					(
+ 						(
+ 							(relationTypeObj.journalDokumentAttr).dokumenttitel IS NULL
+ 							OR
+ 							(a.journal_dokument_attr).dokumenttitel ILIKE (relationTypeObj.journalDokumentAttr).dokumenttitel
+ 						)
+ 						AND
+ 						(
+ 							(relationTypeObj.journalDokumentAttr).offentlighedundtaget IS NULL
+ 							OR
+ 								(
+ 									(
+ 										((relationTypeObj.journalDokumentAttr).offentlighedundtaget).AlternativTitel IS NULL
+ 										OR
+ 										((a.journal_dokument_attr).offentlighedundtaget).AlternativTitel ILIKE ((relationTypeObj.journalDokumentAttr).offentlighedundtaget).AlternativTitel 
+ 									)
+ 									AND
+ 									(
+ 										((relationTypeObj.journalDokumentAttr).offentlighedundtaget).Hjemmel IS NULL
+ 										OR
+ 										((a.journal_dokument_attr).offentlighedundtaget).Hjemmel ILIKE ((relationTypeObj.journalDokumentAttr).offentlighedundtaget).Hjemmel
+ 									)
+ 								)
+ 						)
+ 					)
+ 				)
+ 				AND
  						(
  				(registreringObj.registrering) IS NULL 
  				OR
  				(
  					(
