*** ./generated-files/as_update_tilstand.sql	2016-11-13 16:51:22.000000000 +0100
--- ./patches/as_update_tilstand.org.sql	2016-11-13 16:48:02.000000000 +0100
***************
*** 36,45 ****
--- 36,51 ----
    new_tilstand_registrering tilstand_registrering;
    prev_tilstand_registrering tilstand_registrering;
    tilstand_relation_navn TilstandRelationKode;
    attrEgenskaberObj TilstandEgenskaberAttrType;
    auth_filtered_uuids uuid[];
+   rel_type_max_index_prev_rev int;
+   rel_type_max_index_arr _tilstandRelationMaxIndex[];
+   tilstand_rel_type_cardinality_unlimited tilstandRelationKode[]:=ARRAY['tilstandsvaerdi'::TilstandRelationKode,'begrundelse'::TilstandRelationKode,'tilstandskvalitet'::TilstandRelationKode,'tilstandsvurdering'::TilstandRelationKode,'tilstandsaktoer'::TilstandRelationKode,'tilstandsudstyr'::TilstandRelationKode,'samtykke'::TilstandRelationKode,'tilstandsdokument'::TilstandRelationKode]::TilstandRelationKode[];
+   tilstand_uuid_underscores text;
+   tilstand_rel_seq_name text;
+   tilstand_rel_type_cardinality_unlimited_present_in_argument tilstandRelationKode[];
  BEGIN
  
  --create a new registrering
  
  IF NOT EXISTS (select a.id from tilstand a join tilstand_registrering b on b.tilstand_id=a.id  where a.id=tilstand_uuid) THEN
***************
*** 75,125 ****
  IF relationer IS NOT NULL AND coalesce(array_length(relationer,1),0)=0 THEN
  --raise notice 'Skipping relations, as it is explicit set to empty array. Update note [%]',note;
  ELSE
  
    --1) Insert relations given as part of this update
!   --2) Insert relations of previous registration, taking overlapping virknings into consideration (using function subtract_tstzrange)
  
    --Ad 1)
  
  
  
        INSERT INTO tilstand_relation (
          tilstand_registrering_id,
            virkning,
              rel_maal_uuid,
                rel_maal_urn,
                  rel_type,
!                   objekt_type
        )
        SELECT
          new_tilstand_registrering.id,
            a.virkning,
              a.uuid,
                a.urn,
                  a.relType,
!                   a.objektType
        FROM unnest(relationer) as a
      ;
  
!    
    --Ad 2)
  
    /**********************/
    -- 0..1 relations 
!    
  
!   FOREACH tilstand_relation_navn in array  ARRAY['tilstandsobjekt'::TilstandRelationKode,'tilstandstype'::TilstandRelationKode]::TilstandRelationKode[]
    LOOP
  
      INSERT INTO tilstand_relation (
          tilstand_registrering_id,
            virkning,
              rel_maal_uuid,
                rel_maal_urn,
                  rel_type,
!                   objekt_type
        )
      SELECT 
          new_tilstand_registrering.id, 
            ROW(
              c.tz_range_leftover,
--- 81,216 ----
  IF relationer IS NOT NULL AND coalesce(array_length(relationer,1),0)=0 THEN
  --raise notice 'Skipping relations, as it is explicit set to empty array. Update note [%]',note;
  ELSE
  
    --1) Insert relations given as part of this update
!   --2) Insert relations of previous registration, with index values not included in this update. Please notice that for the logic to work,
!   --  it is very important that the index sequences start with the max value for index of the same type in the previous registration
  
    --Ad 1)
  
+ --build array with the max index values of the different types of relations of the previous registration
+ SELECT array_agg(rel_type_max_index)::_tilstandRelationMaxIndex[] into rel_type_max_index_arr
+ FROM
+ (
+   SELECT
+   (ROW(rel_type,coalesce(max(rel_index),0))::_tilstandRelationMaxIndex) rel_type_max_index  
+   FROM tilstand_relation a
+   where a.tilstand_registrering_id=prev_tilstand_registrering.id
+   and a.rel_type = any (tilstand_rel_type_cardinality_unlimited)
+   group by rel_type
+ ) as a
+ ;
+ 
  
+ --Create temporary sequences
+ 
+ SELECT array_agg( DISTINCT a.RelType) into tilstand_rel_type_cardinality_unlimited_present_in_argument FROM  unnest(relationer) a WHERE a.RelType = any (tilstand_rel_type_cardinality_unlimited) ;
+ tilstand_uuid_underscores:=replace(tilstand_uuid::text, '-', '_');
+ 
+ IF coalesce(array_length(tilstand_rel_type_cardinality_unlimited_present_in_argument,1),0)>0 THEN
+ FOREACH tilstand_relation_navn IN ARRAY (tilstand_rel_type_cardinality_unlimited_present_in_argument)
+   LOOP
+   tilstand_rel_seq_name := 'tilstand_' || tilstand_relation_navn::text || tilstand_uuid_underscores;
+ 
+   rel_type_max_index_prev_rev:=null;
+ 
+   SELECT 
+     a.indeks into rel_type_max_index_prev_rev
+   FROM
+     unnest(rel_type_max_index_arr) a(relType,indeks)
+   WHERE
+     a.relType=tilstand_relation_navn
+   ;
+   
+   IF rel_type_max_index_prev_rev IS NULL THEN
+     rel_type_max_index_prev_rev:=0;
+   END IF;
+ 
+   EXECUTE 'CREATE TEMPORARY SEQUENCE ' || tilstand_rel_seq_name || '
+   INCREMENT 1
+   MINVALUE 1
+   MAXVALUE 9223372036854775807
+   START ' ||  (rel_type_max_index_prev_rev+1)::text ||'
+   CACHE 1;';
+ 
+ END LOOP;
+ END IF;
  
        INSERT INTO tilstand_relation (
          tilstand_registrering_id,
            virkning,
              rel_maal_uuid,
                rel_maal_urn,
                  rel_type,
!                   objekt_type,
!                     rel_index,
!                       tilstand_vaerdi_attr
        )
        SELECT
          new_tilstand_registrering.id,
            a.virkning,
              a.uuid,
                a.urn,
                  a.relType,
!                   a.objektType,
!                     CASE 
!                     WHEN a.relType = any (tilstand_rel_type_cardinality_unlimited) THEN
!                       CASE WHEN a.indeks IS NULL OR b.id IS NULL THEN --for new relations and relations with index given that is not found in prev registrering, we'll assign new index values 
!                         nextval('tilstand_' || a.relType::text || tilstand_uuid_underscores)
!                       ELSE
!                         a.indeks
!                       END
!                     ELSE
!                     NULL
!                     END,
!                       CASE
!                         WHEN 
!                         a.relType='tilstandsvaerdi' AND
!                          ( NOT ((a.tilstandsVaerdiAttr) IS NULL))
!                          AND 
!                          (
!                            (a.tilstandsVaerdiAttr).forventet IS NOT NULL
!                            OR
!                            (a.tilstandsVaerdiAttr).nominelVaerdi IS NOT NULL
!                          ) THEN (a.tilstandsVaerdiAttr)
!                         ELSE
!                         NULL
!                       END
        FROM unnest(relationer) as a
+       LEFT JOIN tilstand_relation b on a.relType = any (tilstand_rel_type_cardinality_unlimited) and b.tilstand_registrering_id=prev_tilstand_registrering.id and a.relType=b.rel_type and a.indeks=b.rel_index
      ;
  
! 
! --Drop temporary sequences
! 
! IF coalesce(array_length(tilstand_rel_type_cardinality_unlimited_present_in_argument,1),0)>0 THEN
! FOREACH tilstand_relation_navn IN ARRAY (SELECT array_agg( DISTINCT a.RelType) FROM  unnest(relationer) a WHERE a.RelType = any (tilstand_rel_type_cardinality_unlimited))
!   LOOP
!   tilstand_rel_seq_name := 'tilstand_' || tilstand_relation_navn::text || tilstand_uuid_underscores;
!   EXECUTE 'DROP  SEQUENCE ' || tilstand_rel_seq_name || ';';
! END LOOP;
! END IF;
! 
    --Ad 2)
  
    /**********************/
    -- 0..1 relations 
!   --Please notice, that for 0..1 relations for tilstand, we're ignoring index here, and handling it the same way, that is done for other object types (like Facet, Klasse etc). That is, you only make changes for the virkningsperiod that you explicitly specify (unless you delete all relations) 
  
!   FOREACH tilstand_relation_navn in array ARRAY['tilstandsobjekt'::TilstandRelationKode,'tilstandstype'::TilstandRelationKode]::TilstandRelationKode[]
    LOOP
  
      INSERT INTO tilstand_relation (
          tilstand_registrering_id,
            virkning,
              rel_maal_uuid,
                rel_maal_urn,
                  rel_type,
!                   objekt_type,
!                     rel_index,
!                       tilstand_vaerdi_attr          
        )
      SELECT 
          new_tilstand_registrering.id, 
            ROW(
              c.tz_range_leftover,
***************
*** 128,138 ****
                (a.virkning).NoteTekst
            ) :: virkning,
              a.rel_maal_uuid,
                a.rel_maal_urn,
                  a.rel_type,
!                   a.objekt_type
      FROM
      (
        --build an array of the timeperiod of the virkning of the relations of the new registrering to pass to _subtract_tstzrange_arr on the relations of the previous registrering 
        SELECT coalesce(array_agg((b.virkning).TimePeriod),array[]::TSTZRANGE[]) tzranges_of_new_reg
        FROM tilstand_relation b
--- 219,231 ----
                (a.virkning).NoteTekst
            ) :: virkning,
              a.rel_maal_uuid,
                a.rel_maal_urn,
                  a.rel_type,
!                   a.objekt_type,
!                     NULL,--a.rel_index, rel_index is not to be used for 0..1 relations    
!                       a.tilstand_vaerdi_attr    
      FROM
      (
        --build an array of the timeperiod of the virkning of the relations of the new registrering to pass to _subtract_tstzrange_arr on the relations of the previous registrering 
        SELECT coalesce(array_agg((b.virkning).TimePeriod),array[]::TSTZRANGE[]) tzranges_of_new_reg
        FROM tilstand_relation b
***************
*** 148,191 ****
      ;
    END LOOP;
  
    /**********************/
    -- 0..n relations
! 
!   --We only have to check if there are any of the relations with the given name present in the new registration, otherwise copy the ones from the previous registration
! 
! 
!   FOREACH tilstand_relation_navn in array ARRAY['tilstandsvaerdi'::TilstandRelationKode,'begrundelse'::TilstandRelationKode,'tilstandskvalitet'::TilstandRelationKode,'tilstandsvurdering'::TilstandRelationKode,'tilstandsaktoer'::TilstandRelationKode,'tilstandsudstyr'::TilstandRelationKode,'samtykke'::TilstandRelationKode,'tilstandsdokument'::TilstandRelationKode]::TilstandRelationKode[]
!   LOOP
! 
!     IF NOT EXISTS  (SELECT 1 FROM tilstand_relation WHERE tilstand_registrering_id=new_tilstand_registrering.id and rel_type=tilstand_relation_navn) THEN
! 
        INSERT INTO tilstand_relation (
              tilstand_registrering_id,
                virkning,
                  rel_maal_uuid,
                    rel_maal_urn,
                      rel_type,
!                       objekt_type
            )
        SELECT 
              new_tilstand_registrering.id,
!               virkning,
!                 rel_maal_uuid,
!                   rel_maal_urn,
!                     rel_type,
!                       objekt_type
!       FROM tilstand_relation
!       WHERE tilstand_registrering_id=prev_tilstand_registrering.id 
!       and rel_type=tilstand_relation_navn 
        ;
  
-     END IF;
-               
-   END LOOP;
- 
- 
  /**********************/
  
  
  END IF;
  /**********************/
--- 241,277 ----
      ;
    END LOOP;
  
    /**********************/
    -- 0..n relations
!   
        INSERT INTO tilstand_relation (
              tilstand_registrering_id,
                virkning,
                  rel_maal_uuid,
                    rel_maal_urn,
                      rel_type,
!                       objekt_type,
!                         rel_index,
!                           tilstand_vaerdi_attr
            )
        SELECT 
              new_tilstand_registrering.id,
!               a.virkning,
!                 a.rel_maal_uuid,
!                   a.rel_maal_urn,
!                     a.rel_type,
!                       a.objekt_type,
!                         a.rel_index,
!                           a.tilstand_vaerdi_attr
!       FROM tilstand_relation a
!       LEFT JOIN tilstand_relation b on b.tilstand_registrering_id=new_tilstand_registrering.id and b.rel_type=a.rel_type and b.rel_index=a.rel_index
!       WHERE a.tilstand_registrering_id=prev_tilstand_registrering.id 
!       and a.rel_type = any (tilstand_rel_type_cardinality_unlimited)
!       and b.id is null --don't transfer relations of prev. registrering, if the index was specified in data given to the/this update-function
        ;
  
  /**********************/
  
  
  END IF;
  /**********************/
