*** ./generated-files/dbtyper-specific_tilstand.sql	2016-10-15 12:52:58.000000000 +0200
--- ./patches/dbtyper-specific_tilstand.org.sql	2016-10-15 12:32:25.000000000 +0200
***************
*** 38,48 ****
  CREATE TYPE TilstandRelationType AS (
    relType TilstandRelationKode,
    virkning Virkning,
    uuid uuid,
    urn  text,
!   objektType text 
  )
  ;
  
  CREATE TYPE TilstandRegistreringType AS
  (
--- 38,49 ----
  CREATE TYPE TilstandRelationType AS (
    relType TilstandRelationKode,
    virkning Virkning,
    uuid uuid,
    urn  text,
!   objektType text,
!   indeks int
  )
  ;
  
  CREATE TYPE TilstandRegistreringType AS
  (
***************
*** 57,63 ****
--- 58,70 ----
  (
    id uuid,
    registrering TilstandRegistreringType[]
  );  
  
+  CREATE Type _TilstandRelationMaxIndex AS
+  (
+    relType TilstandRelationKode,
+    indeks int
+  );
+ 
  
  
