*** ../generated-files/json-cast-functions_dokument.sql	2015-08-05 13:06:04.720885462 +0200
--- ../patches/json-cast-functions_dokument.org.sql	2015-08-05 13:09:44.204882995 +0200
***************
*** 84,93 ****
--- 84,94 ----
    ,CASE WHEN coalesce(array_length($1.relationer,1),0)>0 THEN
      (SELECT rel_json from rel)
      ELSE
      '{}'::json
      END relationer
+   ,$1.varianter  
    FROM
      (
      SELECT
       (SELECT LOWER(($1.registrering).TimePeriod)) as TidsstempelDatoTid 
      ,(SELECT lower_inc(($1.registrering).TimePeriod)) as GraenseIndikator 
