*** ../generated-files/json-cast-functions_dokument.sql	2015-08-04 15:59:45.862987134 +0200
--- ../patches/json-cast-functions_dokument.org.sql	2015-08-05 08:06:55.072456284 +0200
***************
*** 84,93 ****
--- 84,94 ----
    ,CASE WHEN coalesce(array_length($1.relationer,1),0)>0 THEN
      (SELECT rel_json from rel) 
      ELSE
      NULL
      END relationer
+   ,$1.varianter  
    FROM
      (
      SELECT
       (SELECT LOWER(($1.registrering).TimePeriod)) as TidsstempelDatoTid 
      ,(SELECT lower_inc(($1.registrering).TimePeriod)) as GraenseIndikator 
