*** ../generated-files/tbls-specific_dokument.sql	2015-07-31 13:26:33.243854490 +0200
--- ../patches/tbls-specific_dokument.org.sql	2015-07-31 13:25:52.751852474 +0200
***************
*** 385,389 ****
--- 385,725 ----
    (((virkning).notetekst) gin_trgm_ops);
  
  
  
  
+ /**********************************************************************/
+ /*                        dokument variant                            */
+ /**********************************************************************/
+ 
+ 
+ CREATE TABLE dokument_variant
+ (
+  varianttekst text NOT NULL,
+   CONSTRAINT dokument_variant_pkey PRIMARY KEY (varianttekst)
+ )
+ WITH (
+   OIDS=FALSE
+ );
+ ALTER TABLE dokument_variant
+   OWNER TO mox;
+ 
+ 
+ CREATE SEQUENCE dokument_variant_egenskaber_id_seq
+   INCREMENT 1
+   MINVALUE 1
+   MAXVALUE 9223372036854775807
+   START 1
+   CACHE 1;
+ ALTER TABLE dokument_variant_egenskaber_id_seq
+   OWNER TO mox;
+ 
+ 
+ CREATE TABLE dokument_variant_egenskaber
+ (
+   id bigint NOT NULL DEFAULT nextval('dokument_variant_egenskaber_id_seq'::regclass), 
+    varianttekst text not null, 
+    arkivering boolean null, 
+    delvisscannet boolean null, 
+    offentliggoerelse boolean null, 
+    produktion boolean null, 
+    virkning Virkning not null CHECK( (virkning).TimePeriod IS NOT NULL AND not isempty((virkning).TimePeriod) ),
+   dokument_registrering_id bigint not null,
+ CONSTRAINT dokument_variant_egenskaber_pkey PRIMARY KEY (id),
+ CONSTRAINT dokument_variant_egenskaber_forkey_dokument_variant FOREIGN KEY (varianttekst) REFERENCES dokument_variant (varianttekst),
+ CONSTRAINT dokument_variant_egenskaber_forkey_dokumentregistrering  FOREIGN KEY (dokument_registrering_id) REFERENCES dokument_registrering (id) MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
+ CONSTRAINT dokument_variant_egenskaber_exclude_virkning_overlap EXCLUDE USING gist (dokument_registrering_id WITH =,varianttekst WITH =, _composite_type_to_time_range(virkning) WITH &&)
+ )
+ WITH (
+   OIDS=FALSE
+ );
+ ALTER TABLE dokument_variant_egenskaber
+   OWNER TO mox;
+ 
+  
+ CREATE INDEX dokument_variant_egenskaber_pat_varianttekst
+   ON dokument_variant_egenskaber
+   USING gin
+   (varianttekst gin_trgm_ops);
+ 
+ CREATE INDEX dokument_variant_egenskaber_idx_varianttekst
+   ON dokument_variant_egenskaber
+   USING btree
+   (varianttekst); 
+ 
+  
+ 
+ CREATE INDEX dokument_variant_egenskaber_idx_arkivering
+   ON dokument_variant_egenskaber
+   USING btree
+   (arkivering); 
+ 
+  
+ 
+ CREATE INDEX dokument_variant_egenskaber_idx_delvisscannet
+   ON dokument_variant_egenskaber
+   USING btree
+   (delvisscannet); 
+ 
+  
+ 
+ CREATE INDEX dokument_variant_egenskaber_idx_offentliggoerelse
+   ON dokument_variant_egenskaber
+   USING btree
+   (offentliggoerelse); 
+ 
+  
+ 
+ CREATE INDEX dokument_variant_egenskaber_idx_produktion
+   ON dokument_variant_egenskaber
+   USING btree
+   (produktion); 
+ 
+ 
+ 
+ 
+ CREATE INDEX dokument_variant_egenskaber_idx_virkning_aktoerref
+   ON dokument_variant_egenskaber
+   USING btree
+   (((virkning).aktoerref));
+ 
+ CREATE INDEX dokument_variant_egenskaber_idx_virkning_aktoertypekode
+   ON dokument_variant_egenskaber
+   USING btree
+   (((virkning).aktoertypekode));
+ 
+ CREATE INDEX dokument_variant_egenskaber_idx_virkning_notetekst
+   ON dokument_variant_egenskaber
+   USING btree
+   (((virkning).notetekst));
+ 
+ CREATE INDEX dokument_variant_egenskaber_pat_virkning_notetekst
+   ON dokument_variant_egenskaber
+   USING gin
+   (((virkning).notetekst) gin_trgm_ops);
+ 
+ 
+ 
+ /**********************************************************************/
+ /*                        dokument del                                */
+ /**********************************************************************/
+ 
+ CREATE TABLE dokument_del
+ (
+  deltekst text NOT NULL,
+   CONSTRAINT dokument_del_pkey PRIMARY KEY (deltekst)
+ )
+ WITH (
+   OIDS=FALSE
+ );
+ ALTER TABLE dokument_del
+   OWNER TO mox;
+ 
+ 
+ 
+ CREATE SEQUENCE dokument_del_egenskaber_id_seq
+   INCREMENT 1
+   MINVALUE 1
+   MAXVALUE 9223372036854775807
+   START 1
+   CACHE 1;
+ ALTER TABLE dokument_del_egenskaber_id_seq
+   OWNER TO mox;
+ 
+ 
+ 
+ 
+ CREATE TABLE dokument_del_egenskaber
+ (
+   id bigint NOT NULL DEFAULT nextval('dokument_del_egenskaber_id_seq'::regclass), 
+    varianttekst text not null, 
+    deltekst text not null, 
+    indeks int null, 
+    indhold text null, 
+    lokation text null, 
+    mimetype text null, 
+    virkning Virkning not null CHECK( (virkning).TimePeriod IS NOT NULL AND not isempty((virkning).TimePeriod) ),
+   dokument_registrering_id bigint not null,
+ CONSTRAINT dokument_del_egenskaber_pkey PRIMARY KEY (id),
+ CONSTRAINT dokument_del_egenskaber_forkey_dokument_registrering  FOREIGN KEY (dokument_registrering_id) REFERENCES dokument_registrering (id) MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
+ CONSTRAINT dokument_del_forkey_dokument_variant  FOREIGN KEY (varianttekst) REFERENCES dokument_variant (varianttekst) MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
+ CONSTRAINT dokument_del_forkey_dokument_del  FOREIGN KEY (deltekst) REFERENCES dokument_del (deltekst) MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
+ CONSTRAINT dokument_del_egenskaber_exclude_virkning_overlap EXCLUDE USING gist (dokument_registrering_id WITH =,varianttekst WITH =,deltekst WITH =, _composite_type_to_time_range(virkning) WITH &&)
+ )
+ WITH (
+   OIDS=FALSE
+ );
+ ALTER TABLE dokument_del_egenskaber
+   OWNER TO mox;
+ 
+  
+ CREATE INDEX dokument_del_egenskaber_pat_varianttekst
+   ON dokument_del_egenskaber
+   USING gin
+   (varianttekst gin_trgm_ops);
+ 
+ CREATE INDEX dokument_del_egenskaber_idx_varianttekst
+   ON dokument_del_egenskaber
+   USING btree
+   (varianttekst); 
+ 
+ CREATE INDEX dokument_del_egenskaber_pat_deltekst
+   ON dokument_del_egenskaber
+   USING gin
+   (deltekst gin_trgm_ops);
+ 
+ CREATE INDEX dokument_del_egenskaber_idx_deltekst
+   ON dokument_del_egenskaber
+   USING btree
+   (deltekst); 
+ 
+  
+ 
+ CREATE INDEX dokument_del_egenskaber_idx_indeks
+   ON dokument_del_egenskaber
+   USING btree
+   (indeks); 
+ 
+  
+ CREATE INDEX dokument_del_egenskaber_pat_indhold
+   ON dokument_del_egenskaber
+   USING gin
+   (indhold gin_trgm_ops);
+ 
+ CREATE INDEX dokument_del_egenskaber_idx_indhold
+   ON dokument_del_egenskaber
+   USING btree
+   (indhold); 
+ 
+  
+ CREATE INDEX dokument_del_egenskaber_pat_lokation
+   ON dokument_del_egenskaber
+   USING gin
+   (lokation gin_trgm_ops);
+ 
+ CREATE INDEX dokument_del_egenskaber_idx_lokation
+   ON dokument_del_egenskaber
+   USING btree
+   (lokation); 
+ 
+  
+ CREATE INDEX dokument_del_egenskaber_pat_mimetype
+   ON dokument_del_egenskaber
+   USING gin
+   (mimetype gin_trgm_ops);
+ 
+ CREATE INDEX dokument_del_egenskaber_idx_mimetype
+   ON dokument_del_egenskaber
+   USING btree
+   (mimetype); 
+ 
+ 
+ 
+ 
+ CREATE INDEX dokument_del_egenskaber_idx_virkning_aktoerref
+   ON dokument_del_egenskaber
+   USING btree
+   (((virkning).aktoerref));
+ 
+ CREATE INDEX dokument_del_egenskaber_idx_virkning_aktoertypekode
+   ON dokument_del_egenskaber
+   USING btree
+   (((virkning).aktoertypekode));
+ 
+ CREATE INDEX dokument_del_egenskaber_idx_virkning_notetekst
+   ON dokument_del_egenskaber
+   USING btree
+   (((virkning).notetekst));
+ 
+ CREATE INDEX dokument_del_egenskaber_pat_virkning_notetekst
+   ON dokument_del_egenskaber
+   USING gin
+   (((virkning).notetekst) gin_trgm_ops);
+ 
+ 
+ 
+ /****************************************************************************************************/
+ 
+ 
+ 
+ /****************************************************************************************************/
+ 
+ CREATE SEQUENCE dokument_del_relation_id_seq
+   INCREMENT 1
+   MINVALUE 1
+   MAXVALUE 9223372036854775807
+   START 1
+   CACHE 1;
+ ALTER TABLE dokument_del_relation_id_seq
+   OWNER TO mox;
+ 
+ 
+ CREATE TABLE dokument_del_relation
+ (
+   id bigint NOT NULL DEFAULT nextval('dokument_del_relation_id_seq'::regclass),
+   dokument_registrering_id bigint not null,
+   varianttekst text not null, 
+   deltekst text not null,
+   virkning Virkning not null CHECK( (virkning).TimePeriod IS NOT NULL AND not isempty((virkning).TimePeriod) ),
+   rel_maal_uuid uuid NULL, 
+   rel_maal_urn text null,
+   rel_type DokumentdelRelationKode not null,
+   objekt_type text null,
+  CONSTRAINT dokument_del_relation_forkey_dokument_registrering  FOREIGN KEY (dokument_registrering_id) REFERENCES dokument_registrering (id) MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
+  CONSTRAINT dokument_del_relation_pkey PRIMARY KEY (id),
+  CONSTRAINT dokument_del_relation_forkey_dokument_variant  FOREIGN KEY (varianttekst) REFERENCES dokument_variant (varianttekst) MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
+  CONSTRAINT dokument_del_relation_forkey_dokument_del  FOREIGN KEY (deltekst) REFERENCES dokument_del (deltekst) MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
+ -- CONSTRAINT dokument_del_relation_no_virkning_overlap EXCLUDE USING gist (dokument_del_registrering_id WITH =, _as_convert_dokument_del_relation_kode_to_txt(rel_type) WITH =, _composite_type_to_time_range(virkning) WITH &&)  WHERE ( rel_type<>('underredigeringaf'::dokument_delRelationKode )) ,-- no overlapping virkning except for 0..n --relations
+  CONSTRAINT dokument_del_relation_either_uri_or_urn CHECK (NOT (rel_maal_uuid IS NOT NULL AND (rel_maal_urn IS NOT NULL AND rel_maal_urn<>'')))
+ );
+ 
+ 
+ CREATE INDEX dokument_del_relation_idx_rel_maal_obj_uuid
+   ON dokument_del_relation
+   USING btree
+   (rel_type,objekt_type,rel_maal_uuid);
+ 
+ CREATE INDEX dokument_del_relation_idx_rel_maal_obj_urn
+   ON dokument_del_relation
+   USING btree
+   (rel_type,objekt_type,rel_maal_urn);
+ 
+ CREATE INDEX dokument_del_relation_idx_rel_maal_uuid
+   ON dokument_del_relation
+   USING btree
+   (rel_type, rel_maal_uuid);
+ 
+ CREATE INDEX dokument_del_relation_idx_rel_maal_uuid_isolated
+   ON dokument_del_relation
+   USING btree
+   (rel_maal_uuid);
+ 
+ CREATE INDEX dokument_del_relation_idx_rel_maal_urn_isolated
+   ON dokument_del_relation
+   USING btree
+   (rel_maal_urn);
+ 
+ CREATE INDEX dokument_del_relation_idx_rel_maal_urn
+   ON dokument_del_relation
+   USING btree
+   (rel_type, rel_maal_urn);
+ 
+ CREATE INDEX dokument_del_relation_idx_virkning_aktoerref
+   ON dokument_del_relation
+   USING btree
+   (((virkning).aktoerref));
+ 
+ CREATE INDEX dokument_del_relation_idx_virkning_aktoertypekode
+   ON dokument_del_relation
+   USING btree
+   (((virkning).aktoertypekode));
+ 
+ CREATE INDEX dokument_del_relation_idx_virkning_notetekst
+   ON dokument_del_relation
+   USING btree
+   (((virkning).notetekst));
+ 
+ CREATE INDEX dokument_del_relation_pat_virkning_notetekst
+   ON dokument_del_relation
+   USING gin
+   (((virkning).notetekst) gin_trgm_ops);
+ 
